<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.22">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Trabalho de Bioquímica</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../../../../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../../../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../../../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../../../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../../../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../../../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../../../../../">
<link href="../../../../../../../avatar.png" rel="icon" type="image/png">
<script src="../../../../../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../../../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../../../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../../../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../../../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../../../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../../../../../site_libs/quarto-html/quarto-syntax-highlighting-7b4406b7675125bc2ba204020e191172.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../../../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../../../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../../../../../site_libs/bootstrap/bootstrap-4b957b30fffc5edb70387ff4d75203a6.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-S1L73VGCG5"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-S1L73VGCG5', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../../../../../../../styles.css">
</head>

<body class="nav-sidebar floating nav-fixed fullcontent quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../../../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../../../../../logo-bioquanti.png" alt="" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../../../../../index.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../../../../erre_en.html"> 
<span class="menu-text">Reproducible Teaching</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-cheatsheets" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Cheatsheets</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-cheatsheets">    
        <li>
    <a class="dropdown-item" href="../../../../../../../cheatsheet/plotly.pdf">
 <span class="dropdown-text">Plotly</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../../../../../cheatsheet/jmol_cheat.pt.pdf">
 <span class="dropdown-text">Jmol</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../../../../../cheatsheet/r_cheat.pt.pdf">
 <span class="dropdown-text">R</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../../../../../../../producao_en.html"> 
<span class="menu-text">Works</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../../../../contato_en.html"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../../../../../../pt/index.html"> 
<span class="menu-text">PT</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools tools-wide">
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-true"></i></a>
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-true"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Trabalho de Bioquímica</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">JSPlotly &amp; GSPlotly - Interactive Graphics &amp; AI</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/superior/jsplotly/jsplotly_geral_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">JSPlotly - Graphical simulation with JavaScript</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/superior/jsplotly/gsplotly_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GSPlotly - AI code generator for JSPlotly</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/superior/jsplotly/jsptl_bioq_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Simulations in Biophysical Chemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/basico/jsplotly/jsplotly_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Simulations at School</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Jmol - 3D molecules</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/superior/jmol/index_jmol_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Atomic models in Biochemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/basico/Curso/index_jmol_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3D molecules at School</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/basico/Jardim/index_Jardim_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">A Molecule Garden</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">R - Maths and visualization</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/superior/r/index_r_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R in Biophysical Chemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/superior/ScriptRplot/scriptrplot_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">scriptRplot: elegant graphics with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/basico/Curso/index_r_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R &amp; RStudio at School</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/basico/R/index_OIER_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interactive charts and maps at School</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Sisma - dynamic diagrams</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/superior/sisma/index_sisma_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Dynamic flows in Biochemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../../../nivel/basico/sisma/index_sisma_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interactive diagrams at School</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Trabalho de Bioquímica</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="toc" class="level2">
<h2 class="anchored" data-anchor-id="toc">TOC</h2>
<ul>
<li>File #1 - PRIMARY STRUCTURE THE 20 NATURALLY-OCCURRING AMINO ACIDS AND THEIR PROPERTIES</li>
<li>File #2 - SECONDARY STRUCTURE THE ALPHA-HELIX</li>
<li>File #3 - SECONDARY STRUCTURE THE BETA-SHEET</li>
<li>File #4 - TERTIARY STRUCTURE THE FOUR HELIX BUNDLE (HEMERYTHRIN)</li>
<li>File #5 - TERTIARY STRUCTURE THE HELIX-TURN-HELIX MOTIF (LAMBDA REPRESSOR)</li>
<li>File #6 - TERTIARY STRUCTURE THE BETA BARREL (BACTERIOCHLOROPHYLL PROTEIN)</li>
<li>File #7 - TERTIARY STRUCTURE A MEMBRANE SPANNING BETA-BARREL (PORIN)</li>
<li>File #8 - MIXED TERTIARY STRUCTURE MOTIFS THE ZINC FINGER</li>
<li>File #9 - MIXED TERTIARY STRUCTURE MOTIFS THE ALPHA/BETA BARREL (A COMMON MOTIF)</li>
<li>File #10 - MIXED TERTIARY STRUCTURE MOTIF THE ALPHA/BETA FOLD OF IMPORTANT SIGNALING PROTEINS IN THE HUMAN BODY (Ras PROTEIN)</li>
<li>File #11 - QUATERNARY STRUCTURE THE COILED-COIL (LEUCINE ZIPPER)</li>
<li>File #12 - QUATERNARY STRUCTURE ASSOCIATION OF TWO IDENTICAL CHAINS (HIV PROTEASE) File #13 - QUATERNARY STRUCTURE ASSOCIATION OF INEQUIVALENT SUBUNITS AND EVOLUTIONARY CONSERVATION (HEMOGLOBIN) File #14 - QUATERNARY STRUCTURE ASSOCIATION OF THREE INEQUIVALENT SUBUNITS (PHOTOREACTION CENTER PROTEIN) File #15 - QUATERNARY STRUCTURE THE IMMUNOGLOBULIN FOLD (ANTIBODY/ANTIGEN COMPLEX) File #16 - THE DISULFIDE BRIDGE STRUCTURAL STABILIZATION OF PROTEINS File #17 - PROTEIN-DNA INTERACTIONS LAMBDA REPRESSOR File #18 - PROTEIN-DNA INTERACTIONS A ZINC FINGER PROTEIN File #19 - PROTEIN ASSEMBLIES VIRAL PROTEIN COAT</li>
</ul>
<section id="tertiary-structure-the-four-helix-bundle-of-hemerythrin-pdb-1hmo" class="level3">
<h3 class="anchored" data-anchor-id="tertiary-structure-the-four-helix-bundle-of-hemerythrin-pdb-1hmo">TERTIARY STRUCTURE – THE FOUR HELIX BUNDLE OF HEMERYTHRIN (PDB: 1HMO)</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;When 2 or more secondary structure elements group together, they form what is known as ‘tertiary structure motifs’. This kinemage shows the arrangement of four separate helices to form a bundle. You can turn on the rest of the protein to see how the four helices are derived from different regions of one continuous amino acid chain (click on ‘turns’ and ‘rest’). The amino acids which are not involved in forming helices take part in ‘loops’ and ‘turns’. Turn on the sidechains to see which point in toward the middle of the bundle to form the hydrophobic core.</div>
<p>This type of motif can be found in a number of different proteins; here you are looking at hemerythrin (from PDB file, 1HMO). Hemerythrin is a protein which functions in marine invertebrates (like mollusks) the way hemoglobin functions in humans: it binds oxygen and then transports it to where the organism needs it. In this kinemage, you can highlight the region within the bundle where one oxygen molecule binds (2 atoms of oxygen per oxygen molecule; red). Specifically, iron atoms (Fe; blue) held in place by sidechains of the protein, in turn bind to the oxygen (hemoglobin also uses iron for this purpose. Remember that iron-deficiencies lead to anemia, i.e., not enough oxygen can get to the parts of your body that need it).</p>
</section>
<section id="tertiary-structure-the-helix-turn-helix-motif-lambda-repressor---pdb-1lmb" class="level3">
<h3 class="anchored" data-anchor-id="tertiary-structure-the-helix-turn-helix-motif-lambda-repressor---pdb-1lmb">TERTIARY STRUCTURE – THE HELIX-TURN-HELIX MOTIF (LAMBDA REPRESSOR) - PDB: 1LMB</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Another type of tertiary structure which consists primarily of alpha-helices is the ‘helix-turn-helix’ motif. This is one of the common configurations found in proteins which bind to DNA. Typically, the second helix is responsible for the interaction with specific regions of DNA and is known as the ‘recognition helix’. After examining the motif, click on ‘protein’ to display the rest of the protein containing this helix-turn-helix. In this kinemage, you see the N-domain of a protein known as lambda repressor (from PDB file, 1LMB), which is part of the ‘on-off switch’ controlling production of lambda, a bacterial virus. View2 presents another look at the entire protein. In a kinemage to be viewed later, this protein’s interaction with DNA will be studied.</div>
</section>
<section id="tertiary-structure-the-large-beta-sheet-bacteriochlorophyll-protein---pdb-3bcl" class="level3">
<h3 class="anchored" data-anchor-id="tertiary-structure-the-large-beta-sheet-bacteriochlorophyll-protein---pdb-3bcl">TERTIARY STRUCTURE – THE LARGE BETA SHEET (BACTERIOCHLOROPHYLL PROTEIN) - PDB: 3BCL</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This kinemage features extensive beta-sheet structure that makes up nearly an entire protein. You can highlight the two beta-sheets separately; one contains 15 beta-strands and the other contains 5 strands. The macromolecule is the bacteriochlorophyll protein (from PDB file, 3BCL) and as the name suggests, it is associated with chlorophyll. Specifically, there are seven chlorophyll molecules located at the interior of the protein and kept in place by sidechains extending inward from the protein (sidechains are not shown in this graphic). The chlorophyll moieties absorb light energy and utilize it for chemical processes. View2 gives a good look into the interior of the protein. Although they are not represented here, hydrogen bonds are present between each adjacent beta-strand and are responsible for maintaining the integrity of the sheets.</div>
</section>
<section id="tertiary-structure-a-membrane-spanning-beta-barrel-porin---pdb-2por" class="level3">
<h3 class="anchored" data-anchor-id="tertiary-structure-a-membrane-spanning-beta-barrel-porin---pdb-2por">TERTIARY STRUCTURE – A MEMBRANE SPANNING BETA-BARREL (PORIN) - PDB: 2POR</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Porin is another excellent example of a beta-barrel protein, found in the inner membranes of certain bacteria (from PDB file, 2POR). In fact, this protein forms a tunnel through which certain metals, important for cellular function, can pass from one side of the membrane to the other. Membranes have hydrophilic and hydrophobic regions which are necessary to separate two aqueous compartments, and therefore, proteins found within membranes also have distinct hydrophilic and hydrophobic regions. In porin, there are layers of ring-type sidechains (aromatics) which interact with the hydrophobic portion of the membrane. These interactions allow the protein to remain stable in the membrane, facilitating passage of the metals through the membrane-soluble molecule.</div>
<p>A particularly interesting feature of porin is the loop which extends into the center of the protein. This loop acts as a ‘gatekeeper’, allowing only metals of a certain diameter and charge to pass through. Of course, it is the types of sidechain located on this loop which determine which metals can proceed. Here you can see the predominance of negatively-charged aspartate and glutamate residues on the loop. Also, the positively-charged arginine and lysine residues, emanating from the barrel toward the inside of the protein, are believed to be influential on ion transport.</p>
</section>
<section id="mixed-tertiary-structure-motifs-the-zinc-finger-pdb-1zaa" class="level3">
<h3 class="anchored" data-anchor-id="mixed-tertiary-structure-motifs-the-zinc-finger-pdb-1zaa">MIXED TERTIARY STRUCTURE MOTIFS – THE ZINC FINGER (PDB: 1ZAA)</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The zinc finger is a DNA-binding motif which contains one alpha-helix and one antiparallel two-stranded beta-sheet, connected by a short loop. The structure shown in this kinemage is from PDB file, 1ZAA. As the name implies, the finger binds a zinc atom in order to maintain the conformation necessary for proper function. The zinc is held in place by four sidechains of histidines and cysteines. Proteins which contain the zinc finger motif are typically involved in regulating DNA transcription in higher organisms. A protein which possesses three of these motifs, and its binding to DNA, will be examined in another kinemage file.</div>
</section>
<section id="mixed-tertiary-structure-motifs-the-alphabeta-barrel-a-common-motif---pdb-2ypi" class="level3">
<h3 class="anchored" data-anchor-id="mixed-tertiary-structure-motifs-the-alphabeta-barrel-a-common-motif---pdb-2ypi">MIXED TERTIARY STRUCTURE MOTIFS – THE ALPHA/BETA BARREL (A COMMON MOTIF) - PDB: 2YPI</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The alpha/beta barrel consisting of 8 parallel beta-strands surrounded by 8 alpha helices was first observed in a protein called ‘triosephosphate isomerase’; hence, this barrel is commonly known as the TIM barrel. The barrel is built up from consecutive beta-strand/alpha-helix/beta-strand units. At one end of the barrel, a chemical reaction takes place in which a sugar is isomerized in a metabolic pathway found in certain bacteria and yeast. The first kinemage shows how the structure of TIM adjusts to accommodate the binding of an enzyme inhibitor (from PDB file, 2YPI). Specifically, a loop moves downward over the inhibitor, much like a lid.</div>
<p>As you will be able to see in kinemage 2, the alpha/beta barrel motif occurs in many different proteins that have different amino acid sequences and support different chemical reactions. The alpha-carbon traces of four of these proteins can be compared by repeatedly clicking on the ‘ANIMATE’ button to cycle through the structures. The structural similarity between the four proteins should be evident. Aside from TIM, the proteins shown in this kinemage are: ‘amylase’ (responsible for the catalysis of carbohydrates into sugar in animals; from PDB file, 6TAA); ‘flavocytochrome’ (which breaks down lactate to produce energy in yeast cells; from PDB file, 1FCB); and ‘RuBisCO’ (involved in the first step of photosynthesis in plants; from PDB file, 1RUS). These proteins are present in different species and have different functions, yet possess similar structures with the chemical reaction occurring in similar locations. This demonstrates the ability of species to adapt a structure, over the course of evolution, which has proven to be functionally efficient.</p>
</section>
<section id="mixed-tertiary-structure-motif-the-alphabeta-fold-of-important-signaling-proteins-in-the-human-body-ras-protein---pdb-4q21" class="level3">
<h3 class="anchored" data-anchor-id="mixed-tertiary-structure-motif-the-alphabeta-fold-of-important-signaling-proteins-in-the-human-body-ras-protein---pdb-4q21">MIXED TERTIARY STRUCTURE MOTIF – THE ALPHA/BETA FOLD OF IMPORTANT SIGNALING PROTEINS IN THE HUMAN BODY (Ras PROTEIN) - PDB: 4Q21</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Evolutionarily-conserved Ras proteins, found in yeast through human species, are believed to be involved in turning on or off specific processes of cell growth. Mutated proteins of this type have been implicated as the cause of certain cancers, presumably by keeping the cell in a perpetual ‘on’ state resulting in abnormal cell proliferation and tumor formation. In this kinemage, you can examine the structure of the Ras protein, which includes an alpha/beta fold with 6 beta-strands comprising a central beta-sheet surrounded by 5 alpha-helices (from PDB file, 4Q21). You can also turn on a feature called ‘Ribbon’ which presents another way of representating the protein backbone (this looks better when ‘Alpha-C’s’ is turned off). The alpha-helices and beta-strands are highlighted in different colors. Note that Ras binds magnesium (Mg) and a nucleotide (GDP shown here) in order to carry out its function.</div>
</section>
<section id="quaternary-structure-the-coiled-coil-leucine-zipper---pdb-2zta" class="level3">
<h3 class="anchored" data-anchor-id="quaternary-structure-the-coiled-coil-leucine-zipper---pdb-2zta">QUATERNARY STRUCTURE – THE COILED-COIL (LEUCINE ZIPPER) - PDB: 2ZTA</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Quaternary structure is defined as the way in which two or more amino acid chains (polypeptides; subunits) associate to form a working (active) protein. Some of the proteins that you have looked at to this point are fully functional as a single amino acid chain (e.g., hemerythrin &amp; Ras). Proteins such as these are considered ‘monomeric’. Other proteins are only functional when two or more chains associate to form multimeric structures (e.g., TIM - active as a dimer (2 chains); porin - active as a trimer (3 chains)).</div>
<p>The coiled-coil describes the portion of a dimeric structure where two identical helices, one contributed by each chain, wind around each other. In the ‘leucine zipper’, the helices are kept together by favorable sidechain interactions primarily involving leucines. This ‘zipper’ acts as a scaffold for some proteins which bind DNA in higher organisms. In other words, the zipper keeps the protein in a dimeric form, allowing another part of the protein to bind DNA. You will be able to see in Kinemage 1, how the leucines repeat every seventh position and interact hydrophobically with one another between chains (shown here: GCN4; from PDB file, 2ZTA). As in the initial helix you examined, sidechains pointing away from the zipper interface are exposed to aqueous solution and are therefore hydrophilic. You can look at the different sidechains by clicking the ‘inner’ and ‘outer’ sidechain buttons. View2 presents an excellent view through the zipper interface. Be sure to examine both views with all sidechains turned off at some point; this will give you a feel for the ‘coiling’ of the two helices. Kinemage 2 shows the extensive coiling of a bacterial protein called tropomyosin (from PDB file, 2TMA).</p>
</section>
<section id="quaternary-structure-association-of-two-identical-chains-hiv-protease---pdb-5hvp" class="level3">
<h3 class="anchored" data-anchor-id="quaternary-structure-association-of-two-identical-chains-hiv-protease---pdb-5hvp">QUATERNARY STRUCTURE – ASSOCIATION OF TWO IDENTICAL CHAINS (HIV PROTEASE) - PDB: 5HVP</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;The HIV protease is a protein involved in the proliferation of the virus which causes AIDS. It is active as a dimer (two chains, also known as subunits) and, as with the coiled coil, the two chains are equivalent (same amino acid sequence). The structure shown in this kinemage is from PDB file, 5HVP. The interface between the two chains is where the important chemistry takes place. Currently, a particular aspect of AIDS research focuses on creating molecules in the laboratory that resemble, but are structurally different from, the molecule which normally binds at the interface. The goal is to identify a molecule which binds to this ‘active’ site, but does not allow the chemistry to occur. Molecules such as these are called ‘inhibitors’, since they inhibit the proper function of the protein. Successful inhibitors can subsequently be used as drugs to counter the disease.</div>
<p>The start-up view shows the complete dimer of HIV-1 protease complexed with the acetyl-pepstatin inhibitor. The flap regions of the protein (res. 44-57) is highlighted, which undergo the most dramatic conformational changes upon inhibitor binding. The two extended orientations of the inhibitor are included. These orientations are very similar, but not identical.</p>
</section>
<section id="quaternary-structure-association-of-inequivalent-subunits-and-evolutionary-conservation-hemoglobin---pdb-1hho" class="level3">
<h3 class="anchored" data-anchor-id="quaternary-structure-association-of-inequivalent-subunits-and-evolutionary-conservation-hemoglobin---pdb-1hho">QUATERNARY STRUCTURE – ASSOCIATION OF INEQUIVALENT SUBUNITS AND EVOLUTIONARY CONSERVATION (HEMOGLOBIN) - PDB: 1HHO</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Hemoglobin is an oxygen binding and transport protein found in a wide range of species, including humans. It consists of four chains (two inequivalent pairs) which associate with one another to form the active species. Each pair consists of two identical subunits (i.e., there are two A-type chains and two B-type chains; each type contains a ‘globin fold’). Kinemage 1 allows you to compare two normal hemoglobin proteins from species at different points in the evolutionary spectrum (fish and human; from PDB files, 1PBX &amp; 1HHO, respectively). This represents a classic example of ‘evolutionary conservation’, meaning that the 3-D structure and function of the protein has undergone minimal change over the course of evolution. This does not imply that the amino acid sequences are equivalent, or even close; simply that the conformations of the folded proteins are very similar. This is true for hemoglobins from a variety of other species, as well.</div>
<p>One can also compare normal human hemoglobin to the abnormal human hemoglobin responsible for the serious disease known as ‘sickle-cell anemia’ (from PDB file, 1HBS). In the abnormal protein, one sidechain in the B-type subunit is different from the corresponding sidechain in the normal protein, caused by a genetic defect. This subtle change does not alter the overall 3-D structure of the protein, as can be seen by comparison of the two proteins. However, the specific amino acid change (mutation) is significant, as the negatively-charged, hydrophilic residue, glutamate, is replaced by the neutral, hydrophobic residue, valine. This results in the association of the abnormal protein with itself to form long protein strings that distort the shape of red blood cells (sickling).</p>
</section>
<section id="quaternary-structure-association-of-three-inequivalent-subunits-photoreaction-center-protein---pdb-5prc" class="level3">
<h3 class="anchored" data-anchor-id="quaternary-structure-association-of-three-inequivalent-subunits-photoreaction-center-protein---pdb-5prc">QUATERNARY STRUCTURE – ASSOCIATION OF THREE INEQUIVALENT SUBUNITS (PHOTOREACTION CENTER PROTEIN) - PDB: 5PRC</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Photoreaction center protein is a membrane-spanning protein, found in certain bacteria, which harvests light energy for internal utilization. It is comprised of three inequivalent amino acid chains, each contributing a total of eleven alpha-helices which span the width of the organism’s membrane. Five helices are contributed from each of the subunits known as ‘light chain’ and ‘medium chain’, while one comes from the ‘heavy chain’. The helices contain an abundance of hydrophobic sidechains (not displayed) that give the protein stability within the cell membrane, a hydrophobic environment.</div>
</section>
<section id="quaternary-structure-the-immunoglobulin-fold---antibodyantigen-complex-pdb-2hfl" class="level3">
<h3 class="anchored" data-anchor-id="quaternary-structure-the-immunoglobulin-fold---antibodyantigen-complex-pdb-2hfl">QUATERNARY STRUCTURE – THE IMMUNOGLOBULIN FOLD - ANTIBODY/ANTIGEN COMPLEX (pdb: 2HFL)</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Vertebrate species, including humans, have a defense mechanism against parasites such as bacteria and viruses. This mechanism is known as the immune system and part of it (the antibody protein) functions by recognizing and binding to the invading particle (the antigen). Once antibodies, also referred to as immunoglobulins, sufficiently bind to the invading particles, they become recognizable by another type of molecule which then removes the invaders from the host system.</div>
<p>All antibodies have a structural region in common; this is known as the ‘immunoglobulin fold’. The fold consists of two antiparallel beta-sheets packed against each other. Essentially, another area of the antibody differs from one antibody to the next, making each one specific for binding a particular antigen. Kinemage 1 shows the Alpha-Carbon trace of the fold region; the backbone (mainchain) hydrogen bonding scheme is included. The two beta-sheets are highlighted separately to show how they pack together.</p>
<p>Kinemage 2 features the recognition part of the antibody (the variable region – HYHEL-5/FAB) bound to the specific antigen (lysozyme) against which it is fighting. This region is divided into a ‘heavy chain’ and ‘light chain’, based on the number of amino acids contained in each portion. Both chains contain two immunoglobulin folds. The association of these two inequivalent chains creates an antigen recognition site at one end. The heavy and light chains are connected to one another at the opposite end by a disulfide bridge (not shown); this region is connected to the remainder of the antibody structure by another disulfide bridge. The process of recognition occurs by the complementing of sidechains on the surface of the antigen by sidechains in the recognition site of the antibody. Structure derived from PDB file, 2HFL.</p>
</section>
<section id="the-disulfide-bridge-structural-stabilization-of-proteins-pdb-5rsa" class="level3">
<h3 class="anchored" data-anchor-id="the-disulfide-bridge-structural-stabilization-of-proteins-pdb-5rsa">THE DISULFIDE BRIDGE – STRUCTURAL STABILIZATION OF PROTEINS (PDB: 5RSA)</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Ribonuclease A, a small single-subunit enzyme that cuts RNA, is an excellent example of a protein held together by disulfide bridges (from PDB file, 5RSA). Each of the bridges is the result of two cysteine sidechains in close proximity (as a result of protein folding) undergoing covalent attachment through the terminal sulfurs; the chemical reaction can be depicted as follows: -S-H + H-S- =&gt; -S-S-. This reaction occurs under oxidative conditions, typically found in extracellular regions. One or more disulfide bridges can be found to connect domains within a protein (intramolecular bridges). They can also be found in multi-subunit proteins between individual polypeptide chains (intermolecular bridges). The disulfide bridge is critical for maintaining the active conformation in many proteins. In fact, breaking disulfide bonds through reductive processes yields a protein with significantly lower activity.</div>
</section>
<section id="protein-dna-interactions-lambda-repressor-pdb-1lmb" class="level3">
<h3 class="anchored" data-anchor-id="protein-dna-interactions-lambda-repressor-pdb-1lmb">PROTEIN-DNA INTERACTIONS – LAMBDA REPRESSOR (PDB: 1LMB)</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;In this kinemage, you can visualize the DNA-binding portion of lambda repressor protein (containing the helix-turn-helix motif) as it is associated to a double strand of DNA. Note that in order to bind DNA effectively, lambda repressor must associate as a dimer (identical subunits A and B). Each subunit is highlighted in a different color, while the helix-turn-helix portions are shown in darker shades. Structure derived from PDB file, 1LMB.</div>
</section>
<section id="protein-dna-interactions-a-zinc-finger-protein-pdb-1zaa" class="level3">
<h3 class="anchored" data-anchor-id="protein-dna-interactions-a-zinc-finger-protein-pdb-1zaa">PROTEIN-DNA INTERACTIONS – A ZINC FINGER PROTEIN (PDB: 1ZAA )</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;This zinc finger protein, known as Zif-268, contains three identical zinc finger motifs (also termed ‘domains’) within a single amino acid chain. Specifically, each finger contains an alpha-helix, a two-stranded beta-sheet, one zinc ion, and in this case, two cysteine and two histidine metal ligands. The protein, a transcription factor, regulates certain DNA processes by binding in the major groove. Structure derived from PDB file, 1ZAA.</div>
</section>
<section id="protein-assemblies-virus-coats-pdb-4sbv" class="level3">
<h3 class="anchored" data-anchor-id="protein-assemblies-virus-coats-pdb-4sbv">PROTEIN ASSEMBLIES – VIRUS COATS (PDB: 4SBV)</h3>
<div class="line-block">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Viruses exist by infecting normal cells and causing the infected cells to produce more virus particles. To accomplish this, viruses must inject copies of their genetic material (DNA and RNA) into host cells, forcing the cells to make copies of it. Eventually, host cells may die, as they are no longer sustaining their own lives, and the virus particles move on to infect other cells. Fortunately, cells have a variety of defense mechanisms and viruses can infect a limited range of cell types, so the extent of viral infection is not that great.</div>
<p>The region of a virus containing the nucleic acids is surrounded by an extensive protein shell called a ‘capsid’ or ‘coat’. The coats can be a variety of different sizes and shapes; some are rod-shaped, while others are spherical and made up of a symmetric, repetitive protein unit. Kinemage 1 shows the basic unit (trimer) of the protein coat for a plant-infecting virus known as southern bean mosaic virus (SBV). All three proteins (subunits) in the unit are equivalent and in this case, they are arranged in a near-triangular configuration. Structure derived from PDB file, 4SBV.</p>
<p>Kinemage 2 shows the 20 equilateral triangles comprising the highly symmetric icosahedron. The gold triangle represents a unit consisting of 3 subunits in the simplest spherical protein coat of known three-dimensional structure, i.e.&nbsp;satellite tobacco necrosis virus. This means there are 60 subunits in the coat, which is also the minimum number of subunits required to insure icosahedral symmetry. The relative position of one triangular unit within the entire coat for SBV is displayed in magenta. SBV possesses a triangulation number of 3 (i.e., T = 3), meaning that the total number of subunits in the icosahedral shell equals 3 x 60.</p>


</section>
</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>