<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.29">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Proteins</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../../avatar.png" rel="icon" type="image/png">
<script src="../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-0815c480559380816a4d1ea211a47e91.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-70b236579891797f3b1e3a78f64d17d9.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-S1L73VGCG5"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-S1L73VGCG5', { 'anonymize_ip': true});
</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-sidebar floating nav-fixed fullcontent quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../logo-unifal.png" alt="" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../index.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../erre_en.html"> 
<span class="menu-text">Reproducible Teaching</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-cheatsheets" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Cheatsheets</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-cheatsheets">    
        <li>
    <a class="dropdown-item" href="../../../cheatsheet/plotly.pdf">
 <span class="dropdown-text">Plotly</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../cheatsheet/jmol_cheat.pt.pdf">
 <span class="dropdown-text">Jmol</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../cheatsheet/r_cheat.pt.pdf">
 <span class="dropdown-text">R</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../../../producao_en.html"> 
<span class="menu-text">Works</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../contato_en.html"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../../pt/index.html"> 
<span class="menu-text">PT</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools tools-wide">
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-true"></i></a>
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-true"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Proteins</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">JSPlotly &amp; GSPlotly - Interactive Graphics &amp; AI</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">JSPlotly - Graphical simulation with JavaScript</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jsplotly/gsplotly_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GSPlotly - AI code generator for JSPlotly</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jsplotly/jsptl_bioq_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Simulations in Biophysical Chemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/jsplotly/jsplotly_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Simulations at School</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Jmol - 3D molecules</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jmol/index_jmol_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Atomic models in Biochemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Curso/index_jmol_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3D molecules at School</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Jardim/index_Jardim_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">A Molecule Garden</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">R - Maths and visualization</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/r/index_r_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R in Biophysical Chemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/ScriptRplot/scriptrplot_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">scriptRplot: elegant graphics with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Curso/index_r_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R &amp; RStudio at School</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/R/index_OIER_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interactive charts and maps at School</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Sisma - dynamic diagrams</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/sisma/index_sisma_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Dynamic flows in Biochemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/sisma/index_sisma_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interactive diagrams at School</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Proteins</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<!-- Google tag (gtag.js) -->
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-S1L73VGCG5"></script>
<script>
 window.dataLayer = window.dataLayer || [];
 function gtag(){dataLayer.push(arguments);}
 gtag('js', new Date());

 gtag('config', 'G-S1L73VGCG5');
</script>
<section id="whats-on-the-pdb-page-rsbc---protein-data-bank" class="level2">
<h2 class="anchored" data-anchor-id="whats-on-the-pdb-page-rsbc---protein-data-bank"><strong>What’s on the PDB page (RSBC - Protein Data Bank)</strong></h2>
<div class="line-block">The PDB database is the largest <a href="https://www.rcsb.org/">digital repository of proteins</a>, currently containing almost 200,000 macromolecular structures deposited from X-ray diffraction and nuclear magnetic resonance data, and used for research and education. Access the page and see its contents.</div>
<div class="line-block">Access the <a href="https://www.rcsb.org/" target="_blank">PDB</a> page and type <em>1AL1</em> (capital or lower case) in the search field. It is also possible to render the structure in <em>JSmol</em> directly by typing <em>load=1al1</em> in the window that opens below.</div>
<p><a href="https://chemapps.stolaf.edu/jmol/jsmol/simple.htm"><img src="jsmol.png" class="img-fluid" style="width:20.0%"></a></p>
<div class="line-block">If you want a <em>direct link</em> try <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1al1&amp;script=Console" target="_blank">1AL1</a>.</div>
<div class="line-block">Verify that the model loaded <em>on the PDB page</em> refers to a structure called an <em>alpha-helix</em>. There is a variety of information and <em>links</em> to access the structure information, such as the <em>Structure summary</em>, <em>3D View</em>, <em>Sequence</em> tabs, and the <em>Macromolecules</em>, <em>Small molecules</em> (linkers, modified residues), or <em>Experimental data</em> (how the data were obtained) boxes. In <em>Macromolecules</em>, for example, you will see that the structure 1AL1 reveals a peptide with the sequence ELLKKLLEELKG, considering the single-letter abbreviations for amino acids. In the upper right corner, you can download the structure (<em>Download files</em>) in several formats, including the PDB format, which is read by <em>Jmol</em> and other molecular viewers. In the <em>Experiment</em> tab, you can identify how the data was obtained (methodology).</div>
</section>
<section id="whats-in-the-pdb-file" class="level2">
<h2 class="anchored" data-anchor-id="whats-in-the-pdb-file"><strong>What’s in the PDB file</strong></h2>
<div class="line-block">Using the <em>PDB website</em>, <em>Jmol</em> or the <em>JSmol applet</em>, download the <em>1acj.pdb</em> file to your computer or mobile device. Open the file in a simple <em>text editor</em> and observe the header information (<em>header</em>), the compound (<em>compnd</em>), the source (<em>source</em>), the experimental data of origin (<em>expdta</em>), authors, review date, journal in which the structure was published, resolution, and other data.</div>
<div class="line-block">To illustrate the model dynamically, load the file <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1acj&amp;script=Console" target="_blank">1acj.pdb</a> into <em>JSmol</em>.</div>
<div class="line-block">Now, in the <em>text file</em>, observe the information contained in a large table whose first column has the term <em>ATOM</em>. This is the structural information of all the atoms in the compound (note that <em>1acj, acetylcholinesterase</em>, has 4193 atoms in the crystal!)</div>
<div class="line-block">Just below there is another table, this one starting with the column <em>HETAM</em>, and referring to the structural information of a compound formed by heteroatoms, that is, atoms that do not participate in the protein structure, such as a ligand.</div>
<div class="line-block">In these two tables, the last column refers to the type of atom, and the penultimate column refers to the temperature factor (<em>B-factor</em>). This factor expresses high values ​​(&gt;50) when the atom is in high vibration (protein surface, for example, due to contact with the solvent) or low values, when the vibration is reduced (inside the protein, therefore). The column immediately following refers to the degree of occupancy (<em>occupancy</em>) of the atom in the crystal; 1.00 indicates the same conformation for the atoms, while different values ​​indicate multiple conformations.<br>
The three previous columns refer to the <em>X, Y and Z coordinates</em> of each atom, which are used for the three-dimensional construction of the structure and geometric calculations (interatomic distance, for example).</div>
</section>
<section id="d-viewers-of-the-rcsb-page---pdb" class="level2">
<h2 class="anchored" data-anchor-id="d-viewers-of-the-rcsb-page---pdb">3D Viewers of the RCSB Page - PDB</h2>
<div class="line-block">The PDB molecule portal also has viewers for renderable models from the PDB structure file. Check this out on the <a href="https://www.rcsb.org/" target="_blank">RSCB-PDB</a> home page. Type <em>1mbo</em> or any other PDB identifier in the search field and go to the <em>3D View</em> tab. Note that the model is rendered in three dimensions. This is an application from the PDB <em>website</em> itself, and its implementation is based on the <a href="https://molstar.org/" target="_blank">Mol*</a> viewer developed by <em>Molstar</em>.</div>
<div class="line-block">On the other hand, going to the bottom right corner of the page, it is possible to observe the structure rendered by two other programs, in <em>Select a different viewer</em>: <em>NGL (WebGL)</em>, and <em>JSmol</em> itself!</div>
</section>
<section id="general-anatomy-of-a-protein" class="level2">
<h2 class="anchored" data-anchor-id="general-anatomy-of-a-protein"><strong>General Anatomy of a Protein</strong></h2>
<div class="line-block">Open the <em>Console</em> and download the following file from the structure bank, or from the PDB page itself:</div>
<p><a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1lev&amp;script=Console" target="_blank">1lev.pdb</a></p>
<div class="line-block">What structure is this? <em>Domain of porcine kidney fructose-1,6-bisphosphatase</em> complexed to a ligand (see the <em>Console</em> information by <em>Jmol</em>, or by the <em>PDB site</em>, or by the <em>show file</em> command already seen. Although it reflects only one molecule, the model does not allow distinction of peculiarities of the structure. In this sense, try the actions below for better visualization and study, pasting each line in the <em>Console</em> and executing one at a time.</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>cartoon only <span class="co"># renders as "cartoon"</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>color structure <span class="co"># changes the color of the model according to its structure 2a.</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>spin <span class="dv">10</span> <span class="co"># rotates the model at a certain speed</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>spin off <span class="co"># to stop the rotation</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">In any protein structure in an aqueous medium, the hydrophobic amino acid residues tend to be located in the <em>core</em> (interior) of the protein, while the polar ones are on its <em>surface</em>. To test this, visualize the protein by following the code snippet below.</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>spacefill only; select hydrophobic; color green; select positive; color blue; select negative; color red; select aromatic; color magenta <span class="co"># distribute colors to amino acid classes</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>rotate <span class="co"># observe the distribution of charged and aromatic residues</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>slab on; slab <span class="dv">50</span> <span class="co"># perform a sagittal cut in the atomic orbitals (slicing, 50% in this case)</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>slab off <span class="co"># remove the slicing feature</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">To visualize the protein structure with <em>different renderings</em>, try:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>backbone only; backbone <span class="dv">50</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>trace only; trace <span class="dv">100</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="protein-structural-hierarchy" class="level3">
<h3 class="anchored" data-anchor-id="protein-structural-hierarchy"><strong>Protein structural hierarchy</strong></h3>
<div class="line-block">In order to facilitate the observation and study of parts of a protein, its structure is didactically subdivided into <em>sequences</em> (or <em>structures</em>), from primary to quinquennial.</div>
<section id="primary-structure" class="level4">
<h4 class="anchored" data-anchor-id="primary-structure"><strong>Primary structure</strong></h4>
<div class="line-block">The <em>primary structure or sequence</em> of a protein is defined by the sequence of its amino acid residues, numbered from the amino terminal (-NH<span class="math inline">\(_{2}\)</span>) to the carboxy terminal (-COOH).<br>
To view the primary structure of the enzyme under study, type: <em>show sequence</em>.<br>
Knowledge of the primary sequence of a protein allows for a wide variety of predictive structural calculations, including its <em>molecular mass</em>, <em>isoelectric point (pI)</em>, <em>polarity</em> and <em>amphipathicity</em> characteristics, as well as its structural prediction in three-dimensional space.</div>
</section>
<section id="secondary-structure" class="level4">
<h4 class="anchored" data-anchor-id="secondary-structure"><strong>Secondary structure</strong></h4>
<div class="line-block">The secondary structure of proteins is basically formed by <em>helices</em>, <em>folds (sheets)</em> and <em>loops</em>. The identification of each is facilitated by the viewer, as shown above (<em>color structure</em> command). When the loop is formed by less than six residues and causes an abrupt change in the direction of the polypeptide chain, it constitutes a <em>turn</em>.</div>
<div class="line-block">For an interactive study of the <em>secondary structure</em>, load the file <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=8cat&amp;script=Console" target="_blank">8cat.pdb</a>. Since the default initial view in <em>Jmol</em> is <em>ball&amp;stick</em>, try:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>restrict protein;cartoon only; color structure</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">This is bovine <em>catalase</em>, an enzyme with hydroperoxidase activity, commonly present in red blood cells. The <em>default</em> representation (by definition) of the <em>structure</em> coloration gives yellow color to the sheets, red to the helices and white to the turns.</div>
<div class="line-block">To confirm these structures, type:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>show info <span class="co"># information about chains and structure 2a. in the model</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">There are also other types of secondary structures besides those mentioned. To visualize them besides the standard coloring, it is necessary for the program to calculate the secondary structures by an internal algorithm called <strong>DSSP - Define Secondary Structure</strong> . To do so:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>calculate structure</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Note that the protein model has disappeared. To view it again, type or use the arrow keys to return to the cartoon representation and structure coloring command (<em>cartoon only; color structure</em>).</div>
<div class="line-block">Now you can see that, after the calculation by the secondary structure algorithm, the image has undergone changes, with new colors appearing, purple for some helices and blue for some turns. These structures have a <em>coded identification</em> as shown below, and displayed by the Console in each section:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>H <span class="sc">-</span> alpha<span class="sc">-</span>helix</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>E <span class="sc">-</span> extended strand of a beta<span class="sc">-</span>sheet</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>T <span class="sc">-</span> turn with H bond</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>B <span class="sc">-</span> isolated beta<span class="sc">-</span>bridge <span class="fu">residue</span> (only two H bonds)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>G <span class="sc">-</span> <span class="dv">3</span><span class="sc">/</span><span class="dv">10</span> helix</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>I <span class="sc">-</span> pi helix</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>S <span class="sc">-</span> <span class="fu">torsion</span> (minimum change of <span class="dv">70</span> degrees <span class="cf">in</span> the curvature of the structure)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="folds-or-beta-sheets" class="level5">
<h5 class="anchored" data-anchor-id="folds-or-beta-sheets">Folds or beta-sheets</h5>
<div class="line-block">Folds or sheets can be <em>parallel</em> or <em>antiparallel</em>, the former being more common due to the nature of the primary structure. To visualize a folded parallel, load <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=2pec&amp;script=Console" target="_blank">2pec.pdb</a>)</div>
<div class="line-block">Provide a convenient visualization by <em>cartoon</em> or <em>rockets</em>, and a coloring by group:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>cartoon only <span class="co"># or rockets only</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>color group <span class="co"># or Group, or groups, or Groups</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">This <em>coloring scheme</em> stains the protein from the amino terminal (NH<span class="math inline">\(_{2}\)</span>-t) region in blue to the carboxyl terminal (COOH-t) region in red.</div>
<div class="line-block">For a better understanding of the secondary structure, load the file <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=2lyz&amp;script=Console" target="_blank">2lyz.pdb</a> related to <em>lysozyme</em>, a glycan hydrolase. And then:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>show info <span class="co"># observation of structures 2as.</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>cartoon only; color structure</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>calculate structure <span class="co"># algorithm for optimizing structures 2as.</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>cartoon only; color structure <span class="co"># just click the up arrow on the keyboard to retrieve the command line</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>Now...</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>restrict <span class="dv">42-60</span> <span class="co"># restricts the view of beta sheets in the protein to a certain region</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Note that the selected sheet has an <em>antiparallel</em> direction; if it is not clear, change the rendering to:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>rockets only</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>spin <span class="dv">20</span> <span class="co"># spin off to turn off</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>wireframe only; wireframe <span class="dv">50</span>; color cpk</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>calculate hbonds; hbonds <span class="fl">0.2</span> <span class="co"># visualization of H-bonds in the structure</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>zoom <span class="dv">150</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">The same can be done to understand the <em>H-bonds</em> in an alpha helix. To do so, return to the original structure and make the following changes:</div>
<p>Note that the structural codes have appeared in the <em>Console</em> and distinct colors in the image.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>select all; cartoon only; color structure; zoom <span class="dv">0</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>calculate structure</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>cartoon only; color structure <span class="co"># visualization of the 2nd calculated structures</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Now select a <span class="math inline">\(\alpha\)</span>-helix. To do this, see that in the <em>Console</em>, the <span class="math inline">\(\alpha\)</span>-helices are represented by <em>H</em>, as already explained. Choose a region containing H and restrict your observation. You can select one with the region described in the <em>Console</em> (for example: H : A:109_A:114), or click on the beginning and end of a helix you want in the image, checking the position of the amino acid residue in the <em>Console.</em> For example:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>restrict <span class="dv">26-36</span> <span class="co"># restricts the view to a selected alpha-helix</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>wireframe <span class="dv">50</span>; color cpk</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>calculate hbonds</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Rotate the molecule and see how the side chain of the residues is distributed in the structure. An <span class="math inline">\(\alpha\)</span>-helix has 3.6 residues per turn, and a pitch of 0.54 nm (vertical distance between two turns). Finally, observe how the H-bonds are distributed. There are also other types of helices, such as the 3/10 helix, to be explored in the next sections.</div>
</section>
<section id="ramachandran-diagram" class="level5">
<h5 class="anchored" data-anchor-id="ramachandran-diagram">Ramachandran Diagram</h5>
<div class="line-block">In short, the <em>Ramachandran diagram</em> refers to a graph represented on the plane indicating the values ​​of the potential dihedral angles for the formation of <span class="math inline">\(\alpha\)</span>-helix, <span class="math inline">\(\beta\)</span>-sheets and turns, and is widely used in the structural prediction of proteins. To view it, type in the <em>Console</em>:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>load<span class="ot">=</span><span class="dv">9</span>pap <span class="co"># load the model of papain, a proteolytic enzyme from papaya</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>plot rama <span class="co"># or...plot ramachandran</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">The chromatic representation can be interpreted in <em>Jmol</em> with the same color scheme as the original rendering. Thus, <span class="math inline">\(\beta\)</span>-sheets are colored in yellow, <span class="math inline">\(\alpha\)</span>-helices in red, and 3/10 helices in purple. Note the content of beta sheets and <span class="math inline">\(\alpha\)</span>-helices in different quadrants, and depending on the dihedral angles <span class="math inline">\(\phi\)</span> and <span class="math inline">\(\psi\)</span>.<br>
Also note the scattering of white dots outside these regions. Click on some that are outside and identify the amino acid. Not by chance, there are a large number of Gly residues, probably participating in secondary structures of turns. You can generalize this visualization by <em>display gly</em>. And you can also check where these residues are in the three-dimensional structure of the protein by typing <em>model 1</em>. Now, return to the original representation:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>plot rama;</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>display all;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Memorize the region covered by alpha helices (red), and identify the <em>Pro</em> residues:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>display all;</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>display pro</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Note that only one residue of <em>Pro</em> is found in a region predicted for an alpha helix. In practice, <em>Pro</em> does not participate in alpha helices because it has a <em>pyrrolidine ring</em> in the side chain, which blocks the orientation of this structure, a steric hindrance.</div>
<div class="line-block">Other representations of properties are also possible with <em>Jmol</em>, as explained in the <em>Commands</em> chapter, such as <em>mass</em>, <em>hydrophobicity</em>, <em>partial charge</em>. To illustrate a three-dimensional plot of dihedral angles as a function of residue number, for example, type:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>plot properties phi psi resno</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The <em>Proteopedia</em> portal has a very interesting interactive <em>link</em> for learning about the <em>Ramachandran Diagram</em> called <a href="https://proteopedia.org/wiki/index.php/Tutorial:Ramachandran_Plot_Inspection" target="_blank">Ramachandran Plot Inspection</a>.</p>
</section>
<section id="supersecondary-structures" class="level5">
<h5 class="anchored" data-anchor-id="supersecondary-structures">Supersecondary structures</h5>
<div class="line-block">They have this name because they are located in an intermediate organization between secondary and tertiary structures. It is a combination of a few secondary structures with a specific geometric arrangement (<em>motifs</em>). As an initial example, load the file <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1yme&amp;script=Console" target="_blank">1yme.pdb</a>, referring to the enzyme carboxypeptidase.</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>cartoon only;color structure</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>restrict <span class="dv">61-108</span> <span class="co"># visual restriction for a supersecondary structure</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>backbone only; backbone <span class="dv">100</span>; color grey</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>spin <span class="dv">15</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">This structure is formed by a <em>beta-alpha-beta</em> unit. With a little creativity the image seems to resemble a <em>staple</em>. Not surprisingly, its structural definition is “hairpin”. Another hairpin presented by carboxypeptidase, in a different view, can be obtained by: <em>restrict 201-242; rockets only; color structure</em>.</div>
<div class="line-block">Now restrict the image to the other supersecondary structure:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>restrict <span class="dv">73-102</span>; cartoon only; color structure</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Although it is inserted in a supersecondary structure <em>alpha-beta-alpha</em>, it is a <em>motif</em>, a structure that repeats itself over a large extent in proteins. In this case, the motif <em>alpha-turn-alpha</em>, or alpha-alpha hairpin.</div>
<div class="line-block">Another supersecondary structure that is curious even in its name is the <em>beta sandwich</em> presented by the structure <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1bmw&amp;script=Console" target="_blank">1bmw.pdb</a>. Perform the standard <em>cartoon</em> rendering and structure coloring, and note the reason for the name by the juxtaposition of two antiparallel beta sheets.<br>
You can find all the secondary structures and motifs of a protein by searching for a PDB-related site: <a href="http://www.ebi.ac.uk/thornton-srv/databases/cgi-bin/pdbsum/\GetPage.pl?pdbcode=index.html">PDBSum Database</a>. Access it, search for carboxypeptidase (<em>1yme</em>), choose the <em>Protein</em> tab, and see the wealth of carboxypeptidase secondary structures and motifs, all with <em>links</em> to other information and related sites on the Internet.</div>
<div class="line-block">To interact with the Jmol loaded with the carboxypeptidase above, select the <em>beta hairpin</em> structure in the <em>PDB Sum</em> site by clicking on its <em>link</em> in the <em>Protein</em> tab. The opened window informs you that this <em>beta hairpin</em>} goes from residue Val33 to Arg40. Well, identify it by selecting it in Jmol, first marking the protein with a wireframe representation. <strong>Tip: copy and paste the code snippet below into the Jmol/JSmol Console</strong>. Separating the commands with <em>semicolons</em> in successive lines simulates the <em>Script Editor</em> in the <em>Console</em> itself.</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>select all;</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>backbone only;</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>color grey;</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>select <span class="dv">33-40</span>;</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>wireframe only;</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>wireframe <span class="dv">50</span>;</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>color magenta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">For another supersecondary structure, now load the file <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1kt7&amp;script=Console" target="_blank">1kt7.pdb</a>, retinol-binding protein. Visualize it as usual in <em>cartoon.</em> Rotate the structure. The yellow section comprises the supersecondary structure called <em>beta-barrel</em>.<br>
Similarly, there is also the <em>alpha-barrel</em>. To visualize it, load the model <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=3lay&amp;script=Console" target="_blank">3lay.pdb</a>, a sequence of the zinc-binding protein in <em>salmonella</em>. Try the default rendering (structure and coloring) and rotate the molecule.</div>
<div class="line-block">Another very common supersecondary structure that interacts with nucleic acid clefts is called a <em>helix-turn-helix</em>. An example can be found in the DNA-binding protein domain of the <em>lac repressor</em>, <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1lcc&amp;script=Console" target="_blank">1lcc.pdb</a>. Load the structure, provide a suitable visualization, and see how it interacts with the DNA molecule.</div>
<div class="line-block">Several other supersecondary structures can occur in proteins, such as the <em>zinc finger</em>, the <em>leucine zipper</em>, the <em>EF hand</em>, and the <em>greek key</em>.</div>
<div class="line-block">The <em>zinc finger</em> and leucine zipper are commonly found interacting with DNA. Try loading <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1aay&amp;script=Console" target="_blank">1aay.pdb</a> and typing or copying/pasting the following sequence:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>cartoon only;</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>color structure;</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>select hetero and not solvent;</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>spacefill <span class="dv">300</span>;</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>color cpk</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Just to get an idea of the <em>leucine zipper</em>, , load <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1ysa&amp;script=Console" target="_blank">1ysa.pdb</a> and type:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>cartoon only;</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>color structure;</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>select leu;</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>spacefill <span class="dv">300</span>;</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>color cpk</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>select positive;</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>wireframe <span class="dv">100</span>;</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>color lightblue</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">In addition to the amphipathic nature of the helices, their interaction with DNA occurs through positive residues in view of the acidic nature of the double strand.</div>
<div class="line-block">In turn, the so-called <em>EF hand</em> is a <em>motif</em> commonly found in calcium ion-binding proteins, such as <em>calmodulin</em>, <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1cll&amp;script=Console" target="_blank">1cll.pdb</a>. Load the structure, provide the usual visualization and type:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>select hetero and not solvent;</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>spacefill <span class="dv">300</span>;</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>color cpk</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">By clicking on a green sphere, its identification, a calcium atom, appears in the <em>Console</em>. To highlight the motif, type:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>select <span class="dv">117-147</span>;</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>color white</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">As you may have noticed, this motif is similar to the <em>helix-turn-helix</em> motif, but it is distinguished from it by being specific for calcium binding.<br>
The <em>Greek key</em>, another supersecondary structure, comprises an arrangement of antiparallel strands, and can be visualized in <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1a2t&amp;script=Console" target="_blank">1a2t.pdb</a>, a nuclease from <em>staphylococcus</em>. It is a little difficult to observe, so a more specific staining is necessary, as follows:</div>
<p>(perform it after the usual visualization):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>select all;</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>cartoon only;</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>color structure;</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>select <span class="dv">7-19</span>;</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>color red;</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>select <span class="dv">21-28</span>;</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>color red;</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>select <span class="dv">29-36</span>;</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>color green;</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>select <span class="dv">71-77</span>;</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>color blue</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Note that the first and second strands in red are in an antiparallel arrangement to each other. The third strand in green is antiparallel to the first (red) strand, outlining the Greek key. The metaphorical drawing of a Greek key, however, is best compared to its two-dimensional structure found in textbooks.</div>
</section>
</section>
<section id="tertiary-structure" class="level4">
<h4 class="anchored" data-anchor-id="tertiary-structure"><strong>Tertiary Structure</strong></h4>
<div class="line-block">The <em>tertiary structure</em> is composed of a single polypeptide chain in a three-dimensional arrangement. Although there are a virtually infinite number of possibilities for these arrangements, proteins are sometimes classified into groups with common functional or structural similarity, and even phylogenetic similarity.</div>
<section id="conservation" class="level5">
<h5 class="anchored" data-anchor-id="conservation">Conservation</h5>
<div class="line-block">To get an idea of ​​the tertiary structural conservation of proteins, download the structure of the electron transport protein <em>cytochrome c</em>, and from 3 sources: fish, rice, and baker’s yeast. In this case, since these are files loaded from physical media, it is necessary to perform their <em>download</em> in advance in the program’s root directory (or in the one made root, by the command <em>cd ?</em>.</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Downloading the structures:</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>load<span class="ot">=</span><span class="dv">5</span>cyt.pdb;write <span class="dv">5</span>cyt.pdb</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>load<span class="ot">=</span><span class="dv">1</span>ccr.pdb;write <span class="dv">1</span>ccr.pdb</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>load<span class="ot">=</span><span class="dv">1</span>ycc.pdb;write <span class="dv">1</span>ycc.pdb</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Loading the models</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>load files <span class="st">"5cyt.pdb"</span> <span class="st">"1ccr.pdb"</span> <span class="st">"1ycc.pdb"</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Representation of the three structures with different colors and on the same screen:</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>frame<span class="sc">*</span> <span class="co"># all models on the same screen</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>select all;</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>trace only;</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>select <span class="fl">1.1</span>; color white; <span class="co"># cit. c de peixe</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>select <span class="fl">2.1</span>; color green; <span class="co"># cit. c de arroz</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>select <span class="fl">3.1</span>; color blue <span class="co"># cit. c de yeast</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Rotate the structures and see their conservation in different phyla.<br>
Having trouble understanding conservation? <em>Jmol</em> allows an animation of the molecules, useful for structural alignment. To do so, type or copy/paste into the <em>Console</em>:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>compare {<span class="fl">1.1</span>} {<span class="fl">2.1</span>} rotate translate <span class="dv">2</span>; <span class="co"># conjugate rotation and translation between cit. c of fish and rice</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>compare {<span class="fl">3.1</span>} {<span class="fl">2.1</span>} rotate translate <span class="dv">2</span>; <span class="co"># conjugate rotation and translation between cit. c of yeast and rice</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>reset <span class="co"># center the models</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>zoomto <span class="fl">0.5</span> <span class="sc">*</span><span class="fl">1.5</span> <span class="co"># increase the size of the models progressively to 50% more, and at a rate of 0.5s at a time</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>spin <span class="dv">15</span> <span class="co"># rotation</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>delay <span class="dv">10</span>; <span class="co"># 20s waiting time</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>spin off <span class="co"># turn off the rotation</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">If you got this far by running the two code snippets above, congratulations! You have just “run” <em>scripts</em> that automate structural observation and analysis, a great advantage of <em>Jmol</em>.</div>
</section>
<section id="conformational-stability" class="level5">
<h5 class="anchored" data-anchor-id="conformational-stability">Conformational stability</h5>
<div class="line-block">Proteins can be classified as <em>globular</em> and <em>fibrous.</em> To study a globular protein, load papain, <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=9pap&amp;script=Console" target="_blank">9pap.pdb</a>. Omit the solvent visualization and visualize it as wireframe, and rotate the model:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>restrict protein;</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>wireframe only; wireframe <span class="dv">50</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>spin on <span class="co"># to stop, "spin off"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">How many amino acid residues does papain have and what is its primary sequence?</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a> show sequence</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="co"># shows sequence 1a.</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>show info <span class="co"># shows sequence 2a.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">How can a protein stabilize itself conformationally? By <em>covalent forces</em> and <em>weak forces</em>. The first ones reside in the <em>peptide bond</em> present in the structure, in number of <em>n-1</em> amino acid residues, as well as in the proximity of two thiol groups of Cys residues, forming <em>disulfide bonds</em>. These bonds are represented in yellow in the structure, but can be highlighted by typing or pasting the following code snippet:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>trace only;</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>ssbonds on;</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>set ssbonds backbone;</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>ssbonds <span class="dv">100</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">To visualize the residues involved in the bond as well as the remaining free thiols, select only <em>Cys.</em> In papain, all <em>Cys</em> are involved in the formation of the disulfide bond:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>ssbonds off;</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>select cys;</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>wireframe <span class="dv">100</span>;</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>color cpk</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Now visualize the <em>hydrogen bonds</em> contained in the structure:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>restrict protein;</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>trace only; calculate hbonds</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>set hbonds backbone <span class="co"># allows you to associate H-bonds to the carbon skeleton (alternative: "set hbonds sidechain)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">If you manually rotate the molecule you will also notice a distribution of H-bonds between <em>beta sheets</em> and <em>alpha helices</em> of the protein’s secondary structure. To turn off H-bonds: <em>hbonds off</em>.</div>
<div class="line-block">Since every atom interacts with another by <em>dispersion forces</em> or electronic delocalization (<em>van der Waals and London</em>), try visualizing the <em>van der Waals clouds</em> with:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>dots on <span class="co"># "dots off" to remove the VDW surface</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Electrostatic interactions also contribute to protein stabilization, as can be seen by typing:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>trace <span class="dv">10</span>;</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>select charged;</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>wireframe <span class="dv">50</span>;</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>color yellow</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">By rotating the structure, observe that these charged residues are located more on the surface of the protein. Identify the opposite charges of these residues, as well as those of a hydrophobic nature by:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>select positive;color blue;select negative; color red <span class="co"># acidic and basic residues</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>select aromatic; wireframe <span class="dv">50</span>; color green <span class="co"># hydrophobic residues</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">The <em>hydrophobic effect</em> (or hydrophobic interaction) participates as a complementary weak force for the native conformation of the protein, as can be initially observed by the positioning of the aromatic residues identified in the model. Generalizing to all hydrophobic residues, it is possible to perceive their massive contribution to the stability of the protein. It is also possible to discern the aromatic residues among the hydrophobic ones:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>select trp;color magenta;select phe; color pink;select tyr;color aquamarine</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Note that while phenolic <em>Tyr</em> interacts with the solvent, indolic <em>Trp</em> remains mainly in the protein core. :</div>
<div class="line-block">The conformational stability of a protein is therefore dictated by strong and weak forces between its atoms and the solvent, in a three-dimensional arrangement that depends on the rotations allowed around each amino acid residue. By analyzing the <em>dihedral angles phi and psi</em> around the peptide bond (see the chapter on <em>Peptides</em>), it is possible to predict the tertiary structure based on the primary structure of the molecule. More precisely, however, it is possible to predict the secondary structure, as represented by the <em>Ramachandran diagram</em>.</div>
</section>
<section id="prosthetic-groups" class="level5">
<h5 class="anchored" data-anchor-id="prosthetic-groups">Prosthetic groups</h5>
<div class="line-block">Several proteins depend functionally on organic structures of different origin from amino acids, the prosthetic groups (<em>apoprotein</em>+<em>prosthetic group</em> = <em>holoprotein</em>). For example, load the respiratory pigment myoglobin, <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1mbo&amp;script=Console" target="_blank">1mbo.pdb</a>. –&gt;</div>
<div class="line-block">Disappear with the solvent, represent it as a trace and highlight the prosthetic group. To identify the latter, it is necessary to know which group it belongs to in the model: <em>show groups</em>. In this case, the group appears named as <em>hem</em>.</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>restrict protein;</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>trace only;</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>trace <span class="dv">30</span>;</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>select hem;</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>wireframe <span class="dv">100</span>; color cpk</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>spin <span class="dv">20</span>;</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>delay <span class="dv">10</span>;</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>spin off</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">The prosthetic group of myoglobin (and also of hemoglobin and some cytochromes) configures a <em>porphyrin</em> ring or <em>heme group</em>) formed by the juxtaposition of four <em>pyrrole rings</em>, with the <em>Fe</em> atom as the central one for the interaction with molecular oxygen. To highlight this oxygen, look for the group to which it belongs, as done above.</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>select oxy;spacefill <span class="dv">200</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Isolate the heme group of the protein with molecular oxygen, bring it closer, and</div>
<p>observe its structure next to the <span class="math inline">\(O_{2}\)</span> molecule.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">restrict</span> (hem,oxy) and not protein;</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>zoom <span class="dv">0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Note how the structure of the heme group coordinated with oxygen is planar. This is due to the binding of O<span class="math inline">\(_{2}\)</span> with the Fe(III) of the metmyoglobin porphyrin, resulting in Fe(II)-O<span class="math inline">\(_{2}\)</span>, and converging the slightly curved conformation of the ring to planar.</div>
</section>
<section id="fibrous-proteins" class="level5">
<h5 class="anchored" data-anchor-id="fibrous-proteins">Fibrous proteins</h5>
<div class="line-block">Another large group alongside globular proteins is the fibrous proteins. To illustrate this, load the structure of collagen, <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1cgd&amp;script=Console" target="_blank">1cgd.pdb</a>, eliminate the solvent, and visualize the model in line drawing and by coloring the chains:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>restrict protein;</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>trace only;</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>trace <span class="dv">100</span>;</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>color chain</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Rotate the structure and notice the interlacing of 3 strands. Check if the secondary structure matches your observation by <em>show info</em>. Note that there are no turns or bends.</div>
<div class="line-block">Show the amino acid residues by <em>show groups</em>. Note that the molecule is composed only of <em>Gly</em>, <em>Ala</em>, <em>Pro</em>, and <em>Hyp</em> (<em>hydroxyproline</em>). This interlacing comprises the <em>3/10 helices</em> of collagen. Provide the primary sequence by <em>show sequence</em> and note how these residues are distributed.</div>
<div class="line-block">Also note how its stretch is greater than that of alpha helices, restricting the observation to the <em>C</em> chain only: <em>restrict :C</em>. In the 3/10 helix of collagen, the pitch is 0.94 nm (larger, therefore, than for <span class="math inline">\(\alpha\)</span>-helix, 0.54 nm). To check this, simply double-click with the right mouse button on a part of the helix, and drag it to another part, until it covers approximately 0.9 nm.<br>
Finally, see how the H-bonds are distributed in the structure:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>calculate hbonds;</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>hbonds <span class="dv">50</span>;</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>color hbonds yellow;</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>hbonds <span class="fl">0.5</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Now load another fibrous protein, <em>fibroin</em> from the silkworm, <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1slk&amp;script=Console" target="_blank">1slk.pdb</a>. First, observe the structure without the didactic visualization by <em>cartoon</em> or similar. At first glance, do they appear to have alpha helices? Check with <em>show info</em>. Note that there are 15 polypeptide chains in the molecule. To distinguish them, <em>color chain</em>. How do you expect the interaction between the chains to be?</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>calculate hbonds;</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>set hbonds backbone;</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>hbonds <span class="dv">50</span>;</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>color hbonds white</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">What is the amino acid composition (<em>show groups</em>)? Note that the structure is composed of a homopolymer of <em>Ala</em> and <em>Gly</em>, in addition to an acetylated N-terminus (<em>ACE</em>) and a methylated C-terminus (<em>NME</em>) in each chain. Now restrict the view to just four chains. For better visualization, select a white background: (and since the H-bonds are also white: ). Now: . Rotate the structure and observe its parallelism. What secondary structure is this? Type: <em>rockets only</em>. Observe the antiparallel orientation of the represented <span class="math inline">\(\beta\)</span>-sheet section.</div>
<div class="line-block">Now load the main chain of feline pyruvate kinase, <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1pkm&amp;script=Console" target="_blank">1pkm.pdb</a>. Visualize it as usual, rotate the molecule and find out how many domains it has. Note that each domain constitutes a distinct compact unit. To make things easier:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>select <span class="dv">116-219</span>;color blue;</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>select <span class="dv">1-115</span>;color green;</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>select <span class="dv">220-388</span>;color red;</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>select <span class="dv">389-530</span>;color pink</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Sometimes domains can form very similar structures, although in very different proteins. To observe this, load the microbial <em>lactate dehydrogenase</em> <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1ldn&amp;script=Console" target="_blank">1ldn.pdb</a>. If you get the structure information (<em>show info</em>), you will see that the model consists of two crystallographic molecules. Add to the frame (<em>frame</em>) the microbial <em>succinate dehydrogenase</em> <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1emd&amp;script=Console" target="_blank">1emd.pdb</a>. For this additional loading, it is necessary that the structure is already in the root directory of <em>Jmol.</em> To do this, load the structure normally and save it by right-clicking and selecting <em>File, Save</em>. To add the structure to the previous molecule:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>load append <span class="st">"1emd.pdb"</span> <span class="co"># command to add models to the program screen</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Place both molecules in the same frame, visualize them in strips (<em>cartoon</em>), but with different colors, and limit the pyruvate kinase to its A chain, enlarging the final image:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>frame<span class="sc">*</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>select all; cartoon only; select <span class="fl">1.1</span>; color white; select <span class="fl">2.1</span>; color green</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>select <span class="fl">1.1</span>; restrict <span class="sc">:</span>A</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>zoom <span class="dv">0</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Finally, align the domain structures and rotate the molecules to visualize their similarity:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>compare {<span class="fl">2.1</span>} {<span class="fl">1.1</span>} rotate translate <span class="dv">2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Note that, although enzymes of microbial origin act on different substrates, the chains have structurally similar domains</div>
</section>
</section>
<section id="quaternary-structure" class="level4">
<h4 class="anchored" data-anchor-id="quaternary-structure"><strong>Quaternary Structure</strong></h4>
<div class="line-block">This additional level of organization (<em>structural hierarchy</em>) has already been discussed in the previous sections. The quaternary structure comprises proteins with more than one polypeptide chain (collagen, hemoglobin, IgG, for example). There is also the <em>quinquenary structure</em>, which corresponds to supramolecular associations formed by the combination of polypeptide chains with each other (protein complexes), or with nucleic chains (nucleoproteins), or lipid chains (biomembranes).<br>
They are dimers, trimers, and oligomers, in general, each of which can perform similar or very distinct functions among its monomers. Functionally, quaternary proteins sometimes have complex interconnected actions, <em>several active sites</em> exhibiting sequential kinetics, as well as <em>interaction sites with natural ligands</em> to modulate their biological activity.</div>
<section id="hemoglobin" class="level5">
<h5 class="anchored" data-anchor-id="hemoglobin">Hemoglobin</h5>
<div class="line-block">Among the most described quaternary structures is <em>hemoglobin</em>, one of several oxygen transport proteins, such as <em>leghemoglobins</em> (legumes), <em>chlorocruorins</em> (annelids), <em>hemerythrin</em> and <em>hemocyanin</em> (invertebrates). Human hemoglobin can be loaded by the file <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1thb&amp;script=Console" target="_blank">1thb.pdb</a>. Visualize it as a <em>cartoon</em> or <em>trace</em>, and color by structure (<em>color structure</em>).</div>
<div class="line-block">What is the predominant structure? Are there any folds? How many chains does it have? Check with <em>show info</em>.</div>
<div class="line-block">Hemoglobin exists in two distinct <em>conformations</em>}, <em>T</em> (low affinity for O<span class="math inline">\(_{2}\)</span>) and <em>R</em> (high affinity for O<span class="math inline">\(_{2}\)</span>). The equilibrium between these states results in the <em>cooperativity</em> of its binding with the four O<span class="math inline">\(_{2}\)</span> molecules. In this PDB file, hemoglobin (<em>Hb</em>) is bound to O<span class="math inline">\(_{2}\)</span> in only two chains, but crystallized in the <em>T</em> conformation.</div>
<div class="line-block">Unlike myoglobin, which saturates with a single <span class="math inline">\(O_{2}\)</span> molecule with a single affinity (<em>hyperbolic saturation curve</em>), hemoglobin binds to O<span class="math inline">\(_{2}\)</span> with initially increasing affinities, producing a <em>sigmoidal saturation curve</em>. For example, the affinity of the 4th oxygen molecule is about 100 times greater than that of the 1st for hemoglobin.<br>
This cooperativity translates into a conformational change produced in hemoglobin upon binding to O<span class="math inline">\(_{2}\)</span>, from its <em>T</em> state to the <em>R</em> state. Briefly, the motion of the heme plane is transmitted to the F helix by a His (helix F8, proximal histidine His87) bound to the Fe atom. Conformational changes in a unit are transmitted across the interfaces of the <span class="math inline">\(\alpha\)</span><span class="math inline">\(_{1}\)</span>-<span class="math inline">\(\beta\)</span><span class="math inline">\(_{2}\)</span> and <span class="math inline">\(\alpha\)</span><span class="math inline">\(_{2}\)</span>-<span class="math inline">\(\beta\)</span><span class="math inline">\(_{1}\)</span> subunits.<br>
This difference between the affinities of Hb and Mb for O<span class="math inline">\(_{2}\)</span> causes <em>Mb</em> to saturate at a pressure of 2.8 torr (venous capillaries, 30 torr), while <em>Hb</em> saturates at 30 torr (one hundred times higher; pulmonary alveoli - 100 torr).</div>
<div class="line-block">For direct observation of <em>Hb</em>, visualize the model with coloring by cartoon chains, identify and highlight the heme prosthetic group (porphyrin ring):</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>select protein;</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>hide solvent;</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>cartoon only;</span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>color chain;</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>show groups;</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>select hem</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>wireframe <span class="dv">100</span>;color cpk</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>spin <span class="dv">10</span></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>delay <span class="dv">5</span></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>spin off</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Identify the chains by clicking on each one. Also identify any possible ligands by typing <em>show groups</em>. Inositol hexaphosphate (<em>IHP</em>) can then be visualized by:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>select ihp;wireframe <span class="dv">50</span>;color cpk</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">IHP is analogous to <em>2,3-BPG</em> (bisphosphoglycerate) of non-mammalian organisms. Rotate the Hb and notice that the IHP is interacting with the chains in a cleft, near the B and D subunits of the protein.</div>
<div class="line-block">2,3-BPG binds to the <em>T</em> state of Hb but not to the <em>R</em> state, thus reducing the protein’s affinity for oxygen. This allows about 40% free oxygen in venous blood. Fetal hemoglobin does not bind as strongly to 2,3-BPG, allowing a high affinity of O<span class="math inline">\(_{2}\)</span> for the protein.</div>
<div class="line-block">How do the Hb chains interact with each other? Which subunits bind to the IHP?</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>select protein; spacefill</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>label \%c <span class="co"># label the "c" chain of Hb</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">The A and B chains represent an <em>alpha/beta dimer</em>. Note the extensive contacts between these chains, as well as between the C and D chains (also alpha/beta). But notice that there are very few contacts between the A and C chains, as well as between B and D, these pairs formed by alpha chains or beta chains. To visualize how the IHP with its six negative phosphate groups interacts with the protein:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>select positive;color blue</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Now visualize only the <em>C</em> chain, colored from the N-terminal to the C-terminal region, with its heme group, and without the overlapping spheres:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>select all; spacefill off; cartoon only;</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>restrict <span class="sc">:</span>c and not ihp;</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>color group;</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>select hem and not iron and <span class="sc">:</span>c;</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>wireframe <span class="dv">100</span>;color cpk; zoom <span class="dv">0</span>;</span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>select iron and <span class="sc">:</span>c; spacefill <span class="dv">200</span>;color cpk</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">How does molecular oxygen interact with hemoglobin?</div>
<p>“Turn off” the ribbon representation (): .</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>select protein and <span class="sc">:</span>c;trace only;trace <span class="dv">10</span>; color groups <span class="co"># chain "b" with highlighted heme group</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">select</span> (his58,his87,val62,phe43) and <span class="sc">:</span>c <span class="co"># selection of residues close to the heme group</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>define Obind selected <span class="co"># defines the selected residues with a label</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>select Obind; wireframe <span class="dv">50</span>;color cpk <span class="co"># didactically represents the group</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>select oxy and <span class="sc">:</span>c;spacefill <span class="dv">150</span>;color cpk <span class="co"># shows O2 in the structure</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Note the proximity of the side chains of His, Val and Phe to the center of the heme group. It can be better visualized with:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">select</span> (Obind, oxy and <span class="sc">:</span>c); dots <span class="fu">on</span> (<span class="st">"dots off"</span> to omit the VDW clouds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">To make it easier, name the selected residues at the alpha carbon:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>select Obind <span class="sc">:</span><span class="er">*</span>.ca; label <span class="st">"\%n \%R"</span>; color label white</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">The interaction of the ferric iron atom (Fe<span class="math inline">\(^{2+}\)</span>) of the heme group occurs through six ligands, four of which are the tetrapyrrolic nitrogens of the <em>prosthetic group</em> of <em>Hb</em>, one of which is the nitrogen of the imidazole ring of the proximal histidine His87 and the sixth ligand is the molecular oxygen between the Fe atom and the side chain of the distal histidine His58. Rotate the structure and see why His87 is proximal. In addition, Val62 and Phe43 contribute to the hydrophobic environment of interaction with molecular oxygen. In the <em>deoxygenated state</em> the Fe atom of the heme group is only pentacoordinated.</div>
<div class="line-block">Finally, a curiosity: the binding of oxygen displaces the porphyrinic Fe within the tetrapyrrole plane by only <em>0.4 angstroms</em>, enough to displace the alpha helix in which proximal His87 is inserted closer to the heme.</div>
</section>
<section id="other-quaternary-proteins" class="level5">
<h5 class="anchored" data-anchor-id="other-quaternary-proteins">Other quaternary proteins</h5>
<div class="line-block">There are a large number of quaternary proteins, some with dozens of polypeptide chains and/or with a large number of domains. During future discussions on metabolic activity, some of these proteins will be studied in greater depth, but for now, we will only visualize the structural complexity in a few examples.</div>
<div class="line-block">Load an antibody, immunoglobulin G, <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1igt&amp;script=Console" target="_blank">1igt.pdb</a>. Rotate the molecule without changing its view, and you will see that it is a complex protein, in the classic shape of the letter “Y”, with at least three large domains linked by a few residues.</div>
<div class="line-block">Also note that in the <em>Console</em> header, there are several sugars linked to the protein, configuring a <em>glycoprotein</em>. Then render the structure as usual: (<em>cartoon only;color structure</em>). Nominally, its chains are divided into heavy and light, the latter formed by <span class="math inline">\(\beta\)</span> sheets that make contact with the antigen.</div>
<div class="line-block">Now, which chains contact the cell surface? Rotating the structure makes it unclear. However, these domains of the heavy chains are linked to carbohydrates. Therefore, visualize the <em>“trunk of the Y”</em> by:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>select hetero and not solvent;wireframe <span class="dv">50</span>;color cpk</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">Another very complex quaternary structure is the photosystem of <em>Rhodopseudomonas</em>, <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1prc&amp;script=Console" target="_blank">1prc.pdb</a>. Load it and represent it by ribbons and coloring chains: <em>cartoon only;color chain</em>.</div>
<div class="line-block">This photosystem is formed by two identical subunits (light and dark green) and two others (pink and red) linked to various photosynthetic pigments:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>select hetero;wireframe <span class="dv">100</span>;color cpk</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="line-block">For a final example of quaternary structure, the complexity of a bacterial chaperone called <em>GroEl</em>, a protein that helps other proteins fold correctly <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1aon&amp;script=Console" target="_blank">1aon.pdb</a>. Load the structure, provide an appropriate visualization with chain coloring, as before, and only then rotate the molecule to visualize its structural complexity. Proteins under folding interact in the central cavity of GroEl. Chaperones depend on ATP for their action, which can be observed by their identification with <em>show groups</em>:</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>select all; color grey; select adp;wireframe <span class="dv">300</span>;color cpk</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="five-year-structure" class="level4">
<h4 class="anchored" data-anchor-id="five-year-structure"><strong>Five-year structure</strong></h4>
<div class="line-block">An example of the highest level of protein structural complexity, the quinquenary structure can be observed in <em>nucleosomes</em> (nucleoproteins), as in <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1aoi&amp;script=Console" target="_blank">1aoi.pdb</a>. Load the molecule, and visualize it as ribbons and stained by structure.</div>
<div class="line-block">Ribosomes also exhibit a 5-year structure, due to their interaction with nucleic acids. To do so, load the file <a href="https://chemapps.stolaf.edu/jmol/jmol.php?pdbid=1qrs&amp;script=Console" target="_blank">1qrs.pdb</a>, and visualize the structure as ribbons and stained by structure or chain.</div>


</section>
</section>
</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>