<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.22">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Interactive Simulations for Biophysical Chemistry with JSPlotly &amp; GSPlotly</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../../nivel/basico/jsplotly/jsplotly_bas_en.html" rel="next">
<link href="../../../nivel/superior/jsplotly/gsplotly_en.html" rel="prev">
<link href="../../../avatar.png" rel="icon" type="image/png">
<script src="../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-7b4406b7675125bc2ba204020e191172.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-4b957b30fffc5edb70387ff4d75203a6.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-S1L73VGCG5"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-S1L73VGCG5', { 'anonymize_ip': true});
</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../logo-unifal.png" alt="" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../index.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../erre_en.html"> 
<span class="menu-text">Reproducible Teaching</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-cheatsheets" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Cheatsheets</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-cheatsheets">    
        <li>
    <a class="dropdown-item" href="../../../cheatsheet/plotly.pdf">
 <span class="dropdown-text">Plotly</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../cheatsheet/jmol_cheat.pt.pdf">
 <span class="dropdown-text">Jmol</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../cheatsheet/r_cheat.pt.pdf">
 <span class="dropdown-text">R</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../../../producao_en.html"> 
<span class="menu-text">Works</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../contato_en.html"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../../pt/index.html"> 
<span class="menu-text">PT</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools tools-wide">
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-true"></i></a>
    <a href="" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-true"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html">JSPlotly &amp; GSPlotly - Interactive Graphics &amp; AI</a></li><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsptl_bioq_en.html">Simulations in Biophysical Chemistry</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">JSPlotly &amp; GSPlotly - Interactive Graphics &amp; AI</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">JSPlotly - Graphical simulation with JavaScript</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jsplotly/gsplotly_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GSPlotly - AI code generator for JSPlotly</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jsplotly/jsptl_bioq_en.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Simulations in Biophysical Chemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/jsplotly/jsplotly_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Simulations at School</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Jmol - 3D molecules</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/jmol/index_jmol_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Atomic models in Biochemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Curso/index_jmol_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3D molecules at School</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Jardim/index_Jardim_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">A Molecule Garden</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">R - Maths and visualization</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/r/index_r_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R in Biophysical Chemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/ScriptRplot/scriptrplot_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">scriptRplot: elegant graphics with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/Curso/index_r_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R &amp; RStudio at School</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/R/index_OIER_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interactive charts and maps at School</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">Sisma - dynamic diagrams</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/superior/sisma/index_sisma_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Dynamic flows in Biochemistry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../nivel/basico/sisma/index_sisma_bas_en.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interactive diagrams at School</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul class="collapse">
  <li><a href="#instructions" id="toc-instructions" class="nav-link active" data-scroll-target="#instructions">Instructions</a></li>
  <li><a href="#acid-base-equilibrium-and-buffer-systems" id="toc-acid-base-equilibrium-and-buffer-systems" class="nav-link" data-scroll-target="#acid-base-equilibrium-and-buffer-systems"><span class="header-section-number">1</span> Acid-base equilibrium and buffer systems</a></li>
  <li><a href="#net-charge-network-in-peptides" id="toc-net-charge-network-in-peptides" class="nav-link" data-scroll-target="#net-charge-network-in-peptides"><span class="header-section-number">2</span> Net charge network in peptides</a></li>
  <li><a href="#interaction-of-oxygen-with-myoglobin-and-hemoglobin" id="toc-interaction-of-oxygen-with-myoglobin-and-hemoglobin" class="nav-link" data-scroll-target="#interaction-of-oxygen-with-myoglobin-and-hemoglobin"><span class="header-section-number">3</span> Interaction of oxygen with myoglobin and hemoglobin</a></li>
  <li><a href="#bohr-effect-on-hemoglobin-ph" id="toc-bohr-effect-on-hemoglobin-ph" class="nav-link" data-scroll-target="#bohr-effect-on-hemoglobin-ph"><span class="header-section-number">4</span> Bohr effect on hemoglobin (pH)</a></li>
  <li><a href="#isoelectric-point-in-proteins" id="toc-isoelectric-point-in-proteins" class="nav-link" data-scroll-target="#isoelectric-point-in-proteins"><span class="header-section-number">5</span> Isoelectric Point in Proteins</a></li>
  <li><a href="#catalysis-and-enzyme-inhibition" id="toc-catalysis-and-enzyme-inhibition" class="nav-link" data-scroll-target="#catalysis-and-enzyme-inhibition"><span class="header-section-number">6</span> Catalysis and enzyme inhibition</a></li>
  <li><a href="#thermodynamic-stability-of-nucleic-acids" id="toc-thermodynamic-stability-of-nucleic-acids" class="nav-link" data-scroll-target="#thermodynamic-stability-of-nucleic-acids"><span class="header-section-number">7</span> Thermodynamic stability of nucleic acids</a></li>
  <li><a href="#van-der-waals-equation-for-ideal-gases" id="toc-van-der-waals-equation-for-ideal-gases" class="nav-link" data-scroll-target="#van-der-waals-equation-for-ideal-gases"><span class="header-section-number">8</span> Van der Waals equation for ideal gases</a></li>
  <li><a href="#equilibrium-of-atp-production-from-reactants-temperature-and-ph" id="toc-equilibrium-of-atp-production-from-reactants-temperature-and-ph" class="nav-link" data-scroll-target="#equilibrium-of-atp-production-from-reactants-temperature-and-ph"><span class="header-section-number">9</span> Equilibrium of ATP production from reactants, temperature, and pH</a></li>
  <li><a href="#variation-of-gibbs-energy-with-temperature" id="toc-variation-of-gibbs-energy-with-temperature" class="nav-link" data-scroll-target="#variation-of-gibbs-energy-with-temperature"><span class="header-section-number">10</span> Variation of Gibbs energy with temperature</a></li>
  <li><a href="#electrochemical-impedance-spectroscopy-eis" id="toc-electrochemical-impedance-spectroscopy-eis" class="nav-link" data-scroll-target="#electrochemical-impedance-spectroscopy-eis"><span class="header-section-number">11</span> Electrochemical impedance spectroscopy (EIS)</a></li>
  <li><a href="#cyclic-voltammetry" id="toc-cyclic-voltammetry" class="nav-link" data-scroll-target="#cyclic-voltammetry"><span class="header-section-number">12</span> Cyclic voltammetry</a></li>
  <li><a href="#diagrams-and-flowcharts" id="toc-diagrams-and-flowcharts" class="nav-link" data-scroll-target="#diagrams-and-flowcharts"><span class="header-section-number">13</span> Diagrams and flowcharts</a></li>
  <li><a href="#data-analysis" id="toc-data-analysis" class="nav-link" data-scroll-target="#data-analysis"><strong>Data analysis</strong></a></li>
  <li><a href="#user-data-insertion" id="toc-user-data-insertion" class="nav-link" data-scroll-target="#user-data-insertion"><span class="header-section-number">14</span> User data insertion</a></li>
  <li><a href="#uploading-files-for-analysis" id="toc-uploading-files-for-analysis" class="nav-link" data-scroll-target="#uploading-files-for-analysis"><span class="header-section-number">15</span> Uploading files for analysis</a></li>
  <li><a href="#linear-data-adjustment" id="toc-linear-data-adjustment" class="nav-link" data-scroll-target="#linear-data-adjustment"><span class="header-section-number">16</span> Linear data adjustment</a></li>
  <li><a href="#polynomial-regression" id="toc-polynomial-regression" class="nav-link" data-scroll-target="#polynomial-regression"><span class="header-section-number">17</span> Polynomial regression</a></li>
  <li><a href="#multilinear-regression" id="toc-multilinear-regression" class="nav-link" data-scroll-target="#multilinear-regression"><span class="header-section-number">18</span> Multilinear regression</a></li>
  <li><a href="#response-surface-methodology-rsm" id="toc-response-surface-methodology-rsm" class="nav-link" data-scroll-target="#response-surface-methodology-rsm"><span class="header-section-number">19</span> Response Surface Methodology (RSM)</a></li>
  <li><a href="#data-smoothing---spline-and-savitzky-golay-filter" id="toc-data-smoothing---spline-and-savitzky-golay-filter" class="nav-link" data-scroll-target="#data-smoothing---spline-and-savitzky-golay-filter"><span class="header-section-number">20</span> Data smoothing - Spline and Savitzky-Golay filter</a></li>
  <li><a href="#anova-and-tukey-hsd-test" id="toc-anova-and-tukey-hsd-test" class="nav-link" data-scroll-target="#anova-and-tukey-hsd-test"><span class="header-section-number">21</span> ANOVA and Tukey-HSD test</a></li>
  <li><a href="#nonlinear-data-adjustment" id="toc-nonlinear-data-adjustment" class="nav-link" data-scroll-target="#nonlinear-data-adjustment"><span class="header-section-number">22</span> Nonlinear data adjustment</a></li>
  <li><a href="#equation-18" id="toc-equation-18" class="nav-link" data-scroll-target="#equation-18">Equation:</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsplotly_geral_en.html">JSPlotly &amp; GSPlotly - Interactive Graphics &amp; AI</a></li><li class="breadcrumb-item"><a href="../../../nivel/superior/jsplotly/jsptl_bioq_en.html">Simulations in Biophysical Chemistry</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Interactive Simulations for Biophysical Chemistry with JSPlotly &amp; GSPlotly</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="line-block">To illustrate the potential use of <em>JSPlotly</em> for biochemistry and related fields, here are some examples of simulations and data analysis whose graphs are often found in textbooks and related content. To get the most out of each topic, try following the suggestions for <em>parametric manipulation</em> in each topic.</div>
<p><br>
</p>
<section id="instructions" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="instructions">Instructions</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Choose a topic;</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Click on the corresponding graph;</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Click on <span class="st">"Add Plot"</span>;</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Use the mouse <span class="cf">for</span> interactivity and<span class="sc">/</span>or edit the code.</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>Reminder<span class="sc">:</span> the editor uses infinite undo<span class="sc">/</span>redo <span class="cf">in</span> the <span class="fu">code</span> (Ctrl<span class="sc">+</span>Z <span class="sc">/</span> Shift<span class="sc">+</span>Ctrl<span class="sc">+</span>Z)<span class="sc">!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="acid-base-equilibrium-and-buffer-systems" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="acid-base-equilibrium-and-buffer-systems"><span class="header-section-number">1</span> Acid-base equilibrium and buffer systems</h2>
<section id="context" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context">Context:</h4>
<div class="line-block">The example illustrates the addition of a base to a system containing a weak acid and its conjugate base. The titration equation refers to a <em>triprotic acid</em>, the phosphoric acid of the buffer of the same name, but it would also work for many others, such as those present in the Krebs cycle (citrate, isocitrate).</div>
</section>
<section id="equation" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation">Equation:</h4>
<p><span class="math display">\[
fa= \frac{1}{1 + 10^{pKa1 - pH}} + \frac{1}{1 + 10^{pKa2 - pH}} + \frac{1}{1 + 10^{pKa3 - pH}}
\]</span> *Where fa = acid fraction (protonatable groups)<br>
</p>
<p><a href="Eq/jsp_acbase.html" target="_blank"><img src="Eq/tit_acbase.png" class="img-fluid"></a><br>
<!---div class="jsplotly_sup-fundo"> ---></p>
</section>
<section id="suggestion" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="st">"A. Converting the phosphate buffer (triprotic) curve to bicarbonate buffer (diprotic)"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Change the pKa values <span class="cf">for</span> the bicarbonate buffer<span class="sc">:</span> pKa1 <span class="ot">=</span> <span class="fl">6.1</span>, and pKa2 <span class="ot">=</span> <span class="fl">10.3</span>;</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Set a very large value <span class="cf">for</span> <span class="fu">pKa3</span> (e.g., <span class="fl">1e20</span>). </span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Click <span class="st">"add plot."</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>Explanation<span class="sc">:</span> pKa is a term that represents the logarithm of a dissociation <span class="fu">constant</span> (<span class="sc">-</span>log[Ka]). With an extreme value, the denominator becomes equally large, canceling out the term that carries pKa3. In JavaScript and other languages, ‘e’ represents the notation <span class="cf">for</span> power of <span class="fl">10.</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="st">"B. Converting the bicarbonate buffer curve to acetate"</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Simply <span class="cf">repeat</span> the above procedure with pKa1 <span class="ot">=</span> <span class="fl">4.75</span> and eliminate pKa2.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="net-charge-network-in-peptides" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="net-charge-network-in-peptides"><span class="header-section-number">2</span> Net charge network in peptides</h2>
<section id="context-1" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-1">Context:</h4>
<div class="line-block">The code refers to the net charge network present in any sequence of amino acid residues. Here we illustrate <em>angiotensin II</em>, an important peptide for blood pressure regulation and electrolyte balance, whose converting enzyme is associated with the mechanism of cell invasibility by SARS-CoV-2.</div>
</section>
<section id="equation-1" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-1">Equation:</h4>
<div class="line-block">There is no direct equation for this simulation, as the algorithm needs to decide the charge based on the basic or acidic nature of the residue at a given pH value (see the <em>script</em>). Thus:</div>
<p><span class="math display">\[
q =
\begin{cases}
-\dfrac{1}{1 + 10^{pK_a - pH}} &amp; \text{(acid group)} \\\\
\dfrac{1}{1 + 10^{pH - pK_a}} &amp; \text{(basic group)}
\end{cases}
\]</span><br>
</p>
<p>*Where</p>
<ul>
<li>pKa = value of the antilogarithm of base 10 for the acid dissociation equilibrium constant, Ka (or log[Ka]).</li>
</ul>
<p><a href="Eq/jsp_cargaAA.html" target="_blank"><img src="Eq/cargaAA.png" class="img-fluid"></a><br>
<!---div class="jsplotly_sup-fundo"> ---></p>
</section>
<section id="suggestion-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-1">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Select the peptide sequence below and observe the charge distribution<span class="sc">:</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="st">"Ala,Lys,Arg,Leu,Phe,Glu,Cys,Asp,His"</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Simulate the pH condition of the <span class="fu">stomach</span> (<span class="st">"const pH = 1.5"</span>) and check the change <span class="cf">in</span> charges <span class="cf">in</span> the peptide. </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Select a physiological <span class="fu">peptide</span> (e.g., oxytocin), observe its charge <span class="cf">in</span> <span class="fu">blood</span> (pH <span class="fl">7.5</span>), and reflect on its potential <span class="cf">for</span> electrostatic interaction with cellular components.</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="st">"Cys,Tyr,Ile,Gln,Asn,Cys,Pro,Leu,Gly"</span> <span class="sc">-</span> oxytocin</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="interaction-of-oxygen-with-myoglobin-and-hemoglobin" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="interaction-of-oxygen-with-myoglobin-and-hemoglobin"><span class="header-section-number">3</span> Interaction of oxygen with myoglobin and hemoglobin</h2>
<section id="context-2" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-2">Context:</h4>
<div class="line-block">The oxygen molecule can combine with the <em>heme</em> group of myoglobin and hemoglobin in different ways, depending on the cooperativity exhibited in the latter due to its quaternary structure. The following model illustrates this interaction using the <em>Hill equation</em>.</div>
</section>
<section id="equation-2" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-2">Equation:</h4>
<p><span class="math display">\[
Y= \frac{pO_2^{nH}}{p_{50}^{nH} + pO_2^{nH}}
\]</span><br>
</p>
<p><em>Where</em></p>
<p>Y = degree of oxygen saturation in the protein;</p>
<ul>
<li>pO<span class="math inline">\(_{2}\)</span> = oxygen pressure;</li>
<li>p<span class="math inline">\(_{50}\)</span> = oxygen pressure at 50% saturation;</li>
<li>nH = Hill coefficient for the interaction;<br>
</li>
</ul>
<p><a href="Eq/jsp_O2.html" target="_blank"><img src="Eq/O2.png" class="img-fluid"></a><br>
<!---div class="jsplotly_sup-fundo"> ---></p>
</section>
<section id="suggestion-2" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-2">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Run the <span class="fu">application</span> (<span class="st">"add plot"</span>). Note that the value of <span class="st">"nH"</span> <span class="cf">for</span> the Hill constant is <span class="st">"1,"</span> meaning there is no cooperativity effect.</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Now replace the value of <span class="st">"nH"</span> with the Hill coefficient <span class="cf">for</span> hemoglobin, <span class="fl">2.8</span>, and run again<span class="sc">!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="bohr-effect-on-hemoglobin-ph" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="bohr-effect-on-hemoglobin-ph"><span class="header-section-number">4</span> Bohr effect on hemoglobin (pH)</h2>
<section id="context-3" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-3">Context:</h4>
<div class="line-block">Some physiological or pathological conditions can alter the binding affinity of oxygen to hemoglobin, such as temperature, certain metabolites (2,3-BPG), and the hydrogen ion content of the solution.</div>
</section>
<section id="equation-3" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-3">Equation:</h4>
<p><span class="math display">\[
Y(pO_2) = \frac{{pO_2^n}}{{P_{50}^n + pO_2^n}}, \quad \text{with } P_{50} = P_{50,\text{ref}} + \alpha (pH_{\text{ref}} - pH)
\]</span><br>
<em>Where</em>,</p>
<ul>
<li>Y = hemoglobin saturation,</li>
<li>pO<span class="math inline">\(_{2}\)</span> = partial pressure of oxygen (in mmHg),</li>
<li>P<span class="math inline">\(_{50}\)</span> = pressure of O<span class="math inline">\(_{2}\)</span> at which hemoglobin is 50% saturated,</li>
<li>P<span class="math inline">\(_{50,ref}\)</span> = 26 mmHg (standard value),</li>
<li><span class="math inline">\(\alpha\)</span> = 50 (Bohr effect intensity),</li>
<li>pH<span class="math inline">\(_{ref}\)</span> = 7.4,</li>
<li>n = 2.8 = Hill coefficient for hemoglobin.</li>
</ul>
<p><a href="Eq/jspl_bohr.html"><img src="Eq/jspl_bohr.png" class="img-fluid"></a></p>
</section>
</section>
<section id="isoelectric-point-in-proteins" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="isoelectric-point-in-proteins"><span class="header-section-number">5</span> Isoelectric Point in Proteins</h2>
<section id="context-4" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="context-4">Context:</h4>
<div class="line-block">The <em>script</em> for this simulation is based on the distribution of the charge network for a polyelectrolyte, and the identification of the pH value at which this network is zero, i.e., the <em>isoelectric point (or isoionic point), pI</em>. The example uses the primary sequence of lysozyme, a hydrolase that acts to break down the microbial wall.</div>
</section>
<section id="equation-4" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-4">Equation:</h4>
<p><span class="math display">\[
q_{\text{net}}(pH) = \sum_{i=1}^{N} \left[ n_i \cdot q_{B_i} + \frac{n_i}{1 + 10^{pH - pK_{a_i}}} \right]
\]</span><br>
<em>Where</em>,</p>
<ul>
<li>qnet = total net charge;</li>
<li>qB$_{i} = charge of the basic form for residue i (e.g., +1 for Lys, 0 for Asp);</li>
<li>n<span class="math inline">\(_{i}\)</span> = number of groups of residue <em>i</em>.</li>
</ul>
<p><a href="Eq/jsp_pI.html"><img src="Eq/pI.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-3" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="suggestion-3">Suggestion:</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="st">"Finding the pI for other proteins"</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> You can check the titration of any other protein or peptide sequence by simply replacing the primary sequence contained <span class="cf">in</span> the code. An assertive way to perform this replacement involves<span class="sc">:</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>a. Search <span class="cf">for</span> the <span class="st">"FASTA"</span> sequence of the protein <span class="cf">in</span> <span class="fu">NCBI</span> (<span class="st">"https://www.ncbi.nlm.nih.gov/protein/"</span>) <span class="sc">-</span> e.g., <span class="st">"papain"</span>;</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>b. Click on <span class="st">"FASTA"</span> and copy the sequence <span class="dv">1</span>a. obtained;</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>c. Paste the sequence into a website <span class="cf">for</span> residue <span class="fu">quantification</span> (e.g., <span class="st">"https://www.protpi.ch/Calculator/ProteinTool"</span>);</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Replace the sequence <span class="cf">in</span> the code. </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="catalysis-and-enzyme-inhibition" class="level2" data-number="6">
<h2 data-number="6" class="anchored" data-anchor-id="catalysis-and-enzyme-inhibition"><span class="header-section-number">6</span> Catalysis and enzyme inhibition</h2>
<section id="context-5" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-5">Context:</h3>
<div class="line-block">The simulation aims to provide a general equation for enzyme inhibition studies, covering <em>competitive, non-competitive, and competitive (pure or mixed) models</em>, also allowing the study of enzyme catalysis in the absence of an inhibitor.</div>
<section id="equation-5" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="equation-5">Equation:</h4>
<p><span class="math display">\[
v=\frac{Vm*S}{Km(1+\frac{I}{Kic})+S(1+\frac{I}{Kiu})}
\]</span><br>
<em>Where</em></p>
<ul>
<li>S = substrate content for reaction;</li>
<li>Vm = reaction limit speed (in books, maximum speed);</li>
<li>Km = Michaelis-Mentem constant;</li>
<li>Kic = inhibitor dissociation equilibrium constant for competitive model;</li>
<li>Kiu = inhibitor dissociation equilibrium constant for non-competitive model<br>
</li>
</ul>
<p><a href="Eq/jsp_kin_inib.html" target="_blank"><img src="Eq/kin_inib.png" class="img-fluid"></a><br>
<!---div class="jsplotly_sup-fundo"> ---></p>
</section>
</section>
<section id="suggestion-4" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-4">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="st">"A. Enzymatic catalysis in the absence of inhibitor."</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Just run the application with the general equation. Note that the values <span class="cf">for</span> Kic and Kiu are <span class="fu">high</span> (<span class="fl">1e20</span>). Thus, with high <span class="st">"dissociation constants,"</span> the interaction of the inhibitor with the enzyme is irrelevant, returning the model to the classic Michaelis<span class="sc">-</span>Mentem equation.</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try changing the values of Vm and Km, comparing graphs.</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Use the geographic coordinates feature on the icon <span class="fu">bar</span> (<span class="st">"Toggle Spike Lines"</span>) to consolidate the mathematical meaning of Km, as well as to observe the effect of different values on the graph.</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="st">"B. Competitive inhibition model."</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> To observe or compare the Michaelis model with the competitive inhibition model, simply replace the value of Kic with a consistent <span class="fu">number</span> (e.g., <span class="at">Kic=</span> <span class="dv">3</span>).</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="st">"C. Incompetitive inhibition model."</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> The same suggestion above applies to the incompetitive model, this time replacing the value <span class="cf">for</span> Kiu.</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="st">"D. Pure non-competitive inhibition model."</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> In this model, the simulation is performed using equal values <span class="cf">for</span> Kic and Kiu.</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="st">"E. Mixed non-competitive inhibition model."</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> For this model, simply assign different values to Kic and Kiu.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="thermodynamic-stability-of-nucleic-acids" class="level2" data-number="7">
<h2 data-number="7" class="anchored" data-anchor-id="thermodynamic-stability-of-nucleic-acids"><span class="header-section-number">7</span> Thermodynamic stability of nucleic acids</h2>
<section id="context-6" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-6">Context:</h3>
<div class="line-block">Several factors contribute to the thermodynamic stability of biopolymers, either stabilizing or destabilizing them. This simulation deals with a thermal denaturation curve for DNA in the presence or absence of some of these compounds: trehalose (stabilizing osmolyte) and guanidine chloride (destabilizing).</div>
</section>
<section id="equation-6" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-6">Equation:</h3>
<p><span class="math display">\[
y(T) = \frac{1}{1 + e^{-\frac{T - T_m}{\beta}}}
\]</span></p>
<p><em>Where</em>,</p>
<ul>
<li>y(T): fraction of denatured DNA at a given temperature T;</li>
<li>Tm: transition temperature (<em>melting</em>, temperature at which 50% of molecules are double-stranded and 50% are single-stranded);</li>
<li><span class="math inline">\(\beta\)</span>: parameter that adjusts the slope of the curve (affected by trehalose and guanidine).</li>
</ul>
<p><em>Note:</em></p>
<ol type="1">
<li>Trehalose as a stabilizer (reduces <span class="math inline">\(\beta\)</span>);</li>
<li>Guanidine as a denaturant (increases <span class="math inline">\(\beta\)</span>);</li>
</ol>
<p><a href="Eq/jsp_denat_DNA_no_osmol.html" target="_blank"><img src="Eq/denat_DNA.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-5" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-5">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try testing various conditions involved <span class="cf">in</span> the simulation, such as<span class="sc">:</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>a<span class="er">)</span> variation of Tm;</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>b<span class="er">)</span> variation of the <span class="sc">$</span>\beta<span class="sc">$</span> parameter;</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>c<span class="er">)</span> variation of trehalose content;</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>d<span class="er">)</span> variation of guanidine chloride content.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="van-der-waals-equation-for-ideal-gases" class="level2" data-number="8">
<h2 data-number="8" class="anchored" data-anchor-id="van-der-waals-equation-for-ideal-gases"><span class="header-section-number">8</span> Van der Waals equation for ideal gases</h2>
<section id="context-7" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-7">Context:</h3>
<div class="line-block">An adaptation that relates the thermodynamic quantities of pressure, volume, and temperature for ideal gases is the <em>van der Waals equation</em>. In this equation, coefficients are calculated that estimate the existence of a volume and interparticle interactions. Thus, the van der Waals equation corrects the ideal gas equation by considering a term to compensate for intermolecular forces (a/V<span class="math inline">\(^{2}\)</span>) and the available volume, which discounts the volume occupied by the gas molecules themselves.</div>
<div class="line-block">The simulation offers additional interactivity through the presence of sliders for temperature and for the finite volume coefficient (<em>b</em>) and particle interaction coefficient (<em>a</em>).</div>
</section>
<section id="equation-7" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-7">Equation:</h3>
<p><span class="math display">\[
P = \frac{RT}{V - b} - \frac{a}{V^2}
\]</span></p>
<ul>
<li>P = gas pressure (atm);</li>
<li>V = molar volume (L);</li>
<li>T = temperature (K);</li>
<li>R = 0.0821 = ideal gas constant (L·atm/mol·K);</li>
<li>a = intermolecular attraction constant (L<span class="math inline">\(^{2}\)</span>·atm/mol$^{2})</li>
<li>b = excluded volume constant (L/mol)</li>
</ul>
<p><a href="Eq/jsp_vanderWaals.html" target="_blank"><img src="Eq/vanderWaals.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-6" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-6">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try varying the parameters of the equation using the slider <span class="cf">for</span> temperature, as well as <span class="cf">for</span> coefficients <span class="st">"a and b."</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Discuss which of the coefficients has the greatest effect on the curve profile and the reason <span class="cf">for</span> this.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="equilibrium-of-atp-production-from-reactants-temperature-and-ph" class="level2" data-number="9">
<h2 data-number="9" class="anchored" data-anchor-id="equilibrium-of-atp-production-from-reactants-temperature-and-ph"><span class="header-section-number">9</span> Equilibrium of ATP production from reactants, temperature, and pH</h2>
<section id="context-8" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-8">Context:</h3>
<div class="line-block">Intracellular ATP production involves the classic chemical equilibrium relationship between reactants and products as a function of reaction temperature, adjusted to a specific pH value. By varying one or both reactant concentrations or physical-chemical parameters, it is possible to quantify the product using the following thermodynamic reaction:</div>
</section>
<section id="equation-8" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-8">Equation:</h3>
<p><span class="math display">\[
\Delta G = \Delta G^{\circ'} + RT \ln\left(\frac{[\text{ADP}] \cdot [\text{P}_i]}{[\text{ATP}]}\right) + 2{,}303 \cdot RT \cdot n_H \cdot \text{pH}
\]</span></p>
<p><em>Where</em>,</p>
<ul>
<li><span class="math inline">\(\Delta\)</span>G = Gibbs energy of the reaction (positive for spontaneously unfavorable synthesis, kJ/mol);</li>
<li><span class="math inline">\(\Delta\)</span>G<span class="math inline">\(^{o'}\)</span> = 30.5 kJ/mol standard biological Gibbs energy for ATP synthesis;</li>
<li>R = 8.314 J/mol/K (universal gas constant);</li>
<li>T=310 K (physiological temperature);</li>
<li>nH<span class="math inline">\(^{+}\)</span> = 1 (number of protons involved in the reaction);</li>
<li>[ADP], [Pi], [ATP] = molar concentrations of reactants and product</li>
</ul>
<p><a href="Eq/jsp_ATP.html" target="_blank"><img src="Eq/ATP.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-7" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-7">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Change the quantities involved <span class="cf">in</span> the expression and compare with previous visualizations. For example, temperature, pH, and ADP and Pi levels.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="variation-of-gibbs-energy-with-temperature" class="level2" data-number="10">
<h2 data-number="10" class="anchored" data-anchor-id="variation-of-gibbs-energy-with-temperature"><span class="header-section-number">10</span> Variation of Gibbs energy with temperature</h2>
<section id="context-9" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-9">Context:</h3>
<div class="line-block">The Gibbs-Helmholtz relation predicts that Gibbs energy varies nonlinearly with temperature in reactions involving a change in the heat capacity of the system (<span class="math inline">\(\Delta\)</span>Cp). This expanded form of the Gibbs equation for variable heat capacity is found in various biochemical phenomena, such as in the phase transition of biomembrane structures subjected to a compound challenge, or in the conformational change that accompanies the protein structure under heating. \</div>
</section>
<section id="equation-9" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-9">Equation:</h3>
<p><span class="math display">\[
\Delta G(T) = \Delta H^\circ - T\,\Delta S^\circ + \Delta C_p \left(T - T_0 - T \ln\left(\frac{T}{T_0}\right)\right)
\]</span></p>
<p><em>Where</em>,</p>
<ul>
<li><span class="math inline">\(\Delta\)</span>G(T) = Gibbs energy of the reaction at each temperature value, kJ/mol);</li>
<li><span class="math inline">\(\Delta\)</span>H<span class="math inline">\(^{o}\)</span> = standard enthalpy of the reaction at T<span class="math inline">\(_{0}\)</span>, normally 298 K (J/mol);</li>
<li><span class="math inline">\(\Delta\)</span>S<span class="math inline">\(^{o}\)</span> = standard entropy of the reaction at T<span class="math inline">\(_{0}\)</span>;</li>
<li><span class="math inline">\(\Delta\)</span>Cp = variation in the heat capacity of the reaction (J/mol·K), assumed to be constant with temperature;</li>
<li>T = temperature of interest (K);</li>
<li>T<span class="math inline">\(_{0}\)</span> = reference temperature, usually 298 K.</li>
<li>R = 8.314 J/mol/K (universal gas constant);</li>
</ul>
<p><a href="Eq/jsp_gibbs.html" target="_blank"><img src="Eq/gibbs.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-8" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-8">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try varying one or more parameters of the expression;</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Test the behavior of the Gibbs curve at a high reference <span class="fu">temperature</span> (simulation <span class="cf">for</span> extremophile organism);</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Simulate the situation where the variation <span class="cf">in</span> heat capacity is zero.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="electrochemical-impedance-spectroscopy-eis" class="level2" data-number="11">
<h2 data-number="11" class="anchored" data-anchor-id="electrochemical-impedance-spectroscopy-eis"><span class="header-section-number">11</span> Electrochemical impedance spectroscopy (EIS)</h2>
<section id="context-10" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-10">Context:</h3>
<div class="line-block">Electrochemical impedance spectroscopy (EIS) is an analytical technique that measures the response of an electrochemical system to a small amplitude sinusoidal disturbance (e.g., 5-10 mV) applied over a wide frequency range (e.g., 100 kHz-1 mHz). EIS provides information about interfacial processes, such as charge transfer, ion diffusion, and passive film formation, by analyzing the complex impedance of the system. It is a non-destructive technique widely used in areas such as corrosion, batteries, fuel cells, biosensors, and protective coatings.</div>
</section>
<section id="equation-10" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-10">Equation:</h3>
<div class="line-block">An electrochemical impedance (EIS) model with typical equivalent circuit components constitutes the modified Randles model: series resistor <em>Rs</em>, a resistor <em>Rp</em> in parallel with a <em>CPE</em> (Constant Phase Element), and a <em>Warburg</em> element. A common representation for EIS is the <em>Nyquist</em> plot, where the real axis represents resistance (Zre) and the imaginary axis represents negative reactance (−Zim). The formalism involved is in the equation below:</div>
<p><span class="math display">\[
Z_{\text{total}}(\omega) = R_s + \left[ \left( \frac{1}{R_p} + Q (j\omega)^n \right)^{-1} \right] + \frac{\sigma}{\sqrt{\omega}} (1 - j)
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>Z<span class="math inline">\(_{total}\)</span> = total impedance at a given frequency;</li>
<li><span class="math inline">\(\omega\)</span> = angular frequency (rad/s), <span class="math inline">\(\omega\)</span>=2πf or <span class="math inline">\(\omega\)</span>=2πf;</li>
<li>R<span class="math inline">\(_{p}\)</span> = ohmic resistance (resistance of the electrolytic solution, wires, contacts, etc.);</li>
<li>R<span class="math inline">\(_{p}\)</span> = Polarization resistance (associated with charge transfer processes, such as electrochemical reactions);</li>
<li><em>Q</em> = Constant associated with the constant phase element (CPE), replaces an ideal capacitor to represent non-ideal behaviors;</li>
<li><em>n</em> = CPE exponent, between 0 and 1; defines the degree of ideality of the capacitive behavior (n = 1: ideal capacitor; n &lt; 1: dispersion);</li>
<li><span class="math inline">\(\sigma\)</span> = Warburg coefficient, associated with ion diffusion in the electrochemical system;</li>
<li><em>j</em> = Imaginary unit, j<span class="math inline">\(^{2}\)</span> = −1;</li>
</ul>
<div class="line-block">An electrochemical impedance model (EIS) with typical equivalent circuit components constitutes the modified Randles model: series resistor <em>Rs</em>, a resistor <em>Rp</em> in parallel with a <em>CPE</em> (Constant Phase Element), and a <em>Warburg</em> element. A common representation for EIS is the <em>Nyquist</em> plot, where the real axis represents resistance (Zre) and the imaginary axis represents negative reactance (−Zim).</div>
<p><a href="Eq/jsp_EIS.html" target="_blank"><img src="Eq/EIS.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-9" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-9">Suggestion</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Check the effect of Rs on the graph by setting its value to <span class="fu">zero</span> (solution resistance);</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Observe the deformation of the semicircle by varying the values of the constant phase <span class="fu">element</span> (e.g., <span class="at">Q =</span> <span class="fl">1e-3</span>; <span class="at">n =</span> <span class="fl">0.6</span> <span class="sc">-</span> dispersion of capacitive behavior);</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try combining other values of the code header parameters to highlight common situations <span class="cf">in</span> electroanalysis<span class="sc">:</span> Rs, Rp, Q, n, and sigma;</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Reduce the Warburg model with constant phase element to a simple Randles model, consisting only of two resistors <span class="cf">in</span> series, the second <span class="cf">in</span> parallel with an ideal capacitor, and without Warburg <span class="fu">diffusion</span> (<span class="at">sigma =</span> <span class="dv">0</span>; <span class="at">n =</span> <span class="dv">1</span>).</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="cyclic-voltammetry" class="level2" data-number="12">
<h2 data-number="12" class="anchored" data-anchor-id="cyclic-voltammetry"><span class="header-section-number">12</span> Cyclic voltammetry</h2>
<section id="context-11" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-11">Context:</h3>
<div class="line-block">Cyclic voltammetry (CV) is an electroanalytical technique for characterizing interfacial (electrode-solution) redox processes. When a standard analyte in quantitative equilibrium between the oxidized and reduced forms is used, the resulting graph acquires an unusual profile, called a <em>duck shape</em>. This results from the peculiar shape obtained when a bidirectional potentiometric scan is performed on a set of electrodes by a potentiostat, resulting in cycles of oxidation and subsequent reduction (application of positive potentials followed by negative potentials). The shape, value, and distance between the resulting peaks allow the analyte to be identified and characterized, as well as the redox processes involved in the experiment.</div>
</section>
<section id="equation-11" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-11">Equation:</h3>
<div class="line-block">Although the formalism for cyclic voltammetry is complex and involves the solution of <em>ordinary differential equations</em> (as well as electroanalytical techniques in general), which is beyond the scope of this material, the resulting current flow can be summarized by the <em>Buttler-Volmer equation</em> below:</div>
<p><span class="math display">\[
j = j_0 \left[ \exp\left(\frac{\alpha n F (E - E^0)}{RT}\right) - \exp\left(\frac{-(1 - \alpha) n F (E - E^0)}{RT}\right) \right]
\]</span> <em>Where:</em></p>
<ul>
<li>j = current density;</li>
<li>j<span class="math inline">\(_{0}\)</span> = exchange current;</li>
<li><span class="math inline">\(\alpha\)</span> = charge transfer coefficient;</li>
<li>E = potential applied to the electrode;</li>
<li>E<span class="math inline">\(^{0}\)</span> = standard electrode potential;</li>
<li>n = number of electrons;</li>
<li>F = Faraday constant (96485 C·mol⁻¹)</li>
<li>R = universal gas constant (8.314 J·mol⁻¹·K⁻¹);</li>
<li>T = temperature<br>
</li>
</ul>
<div class="line-block">For a model of electroanalyte diffusion and accumulation at the electrode surface, the general equation for the boundary condition where the redox conversion rate is proportional to the diffusion current can be represented by <em>Fick’s 2nd Law equation </em> for mass transport:</div>
<p><span class="math display">\[
\frac{\partial C(x,t)} {\partial t} = D \frac{\partial^2 C(x,t)} {\partial x^2}
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>C(x,t) = concentration of the electroanalytical species (mol/cm³), as a function of position <em>x</em> and time <em>t</em>;</li>
<li>D = diffusion coefficient of the species (cm²/s);</li>
<li>t = time (s);</li>
<li>x = distance from the electrode surface (cm);</li>
<li><span class="math inline">\(\delta\)</span> = notation for partial derivative.<br>
</li>
</ul>
<div class="line-block">The chemical equilibrium of the redox reaction is given by the ratio between the concentrations of the oxidized and reduced forms at the electrode/solution interface when the system is in equilibrium (or near equilibrium), according to the <em>Nernst equation</em>:</div>
<p><span class="math display">\[
E = E^0 + \frac{RT}{nF} \ln \left( \frac{[\text{Ox}]}{[\text{Red}]} \right)
\]</span></p>
<div class="line-block">The resulting current is obtained by <em>Faraday’s law</em>:</div>
<p><span class="math display">\[
i(t) = n F A D \left. \frac{\partial C(x,t)} {\partial x} \right|_{x=0}
\]</span> <em>Where:</em></p>
<ul>
<li>A = electrode surface area (cm²);</li>
<li>i(t) = current at time <em>t</em> (A, Amperes).</li>
</ul>
</section>
<section id="cyclic-voltammetry-simulator" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="cyclic-voltammetry-simulator">Cyclic Voltammetry Simulator</h3>
<div class="line-block">To illustrate the potential of the <em>JSPlotly/GSPlotly</em> platform for developing complex <em>virtual experiments</em>, here is a <em>cyclic voltammetry simulator</em> resulting from the above procedures (as well as a few hours of prompt engineering!!).</div>
<div class="line-block">The example illustrates typical behavior in cyclic voltammetry (and the curious <em>duck shape</em>), and includes a model of diffusion and accumulation of electroactive species near the electrode surface. The code uses a <em>numerical solution</em> to obtain the current density over time, considering the effects of cyclic scanning of the applied potential, mass transport, and reaction kinetics.</div>
<div class="line-block">In short, it numerically solves Fick’s equation with dynamic boundary conditions, applies the Nernst equation at the boundary (electrode surface) to determine the conversion between species, uses the finite difference method or explicit scheme to advance in time, and obtains the current by applying Faraday’s Law, through the derivative of the flux at the interface.</div>
<div class="line-block">Find it complex? How about this: the code combines Fick (transport) and Nernst (local redox equilibrium) to accurately model the shape of the voltamogram, obtaining a physical simulation with significant realism for cyclic voltammetry.</div>
<p><br>
</p>
<p><a href="Eq/jsp_cv.html" target="_blank"><img src="Eq/cv.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-10" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-10">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Observe the number of values at the beginning of the code, which are tangible to <span class="st">"parametric manipulation."</span> Try to understand what they represent, and consciously vary their values, aiming to add value to the learning of the simulation. This is the essence of <span class="st">"parametric manipulation"</span> that involves <span class="st">"reproducible teaching"</span><span class="sc">!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="diagrams-and-flowcharts" class="level2" data-number="13">
<h2 data-number="13" class="anchored" data-anchor-id="diagrams-and-flowcharts"><span class="header-section-number">13</span> Diagrams and flowcharts</h2>
<section id="context-12" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-12">Context:</h3>
<div class="line-block">Due to the nature of the <em>Plotly.js</em> library, it is also possible to generate code in <em>JSPlotly</em> without involving an equation, such as for representing diagrams, flowcharts, heredograms, among others. The example below illustrates a representation of the Krebs cycle.</div>
<p><a href="Eq/jsp_krebs.html" target="_blank"><img src="Eq/krebs.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-11" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-11">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try repositioning enzymes and metabolites by clicking and dragging the terms;</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try replacing the names <span class="cf">in</span> the code to produce another metabolic cycle, such as the urea cycle.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="context-flowchart" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-flowchart">Context: Flowchart</h3>
<p><a href="Eq/jsp_fluxograma.html" target="_blank"><img src="Eq/fluxograma.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-12" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-12">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Reposition terms and connectors by dragging with the mouse;</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> For a different flowchart <span class="cf">in</span> the content, modify the terms <span class="cf">in</span> the code;</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> For a different flowchart format, change the font and connector characteristics <span class="cf">in</span> the <span class="st">"annotations"</span> constant.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="data-analysis" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="data-analysis"><strong>Data analysis</strong></h2>
</section>
<section id="user-data-insertion" class="level2" data-number="14">
<h2 data-number="14" class="anchored" data-anchor-id="user-data-insertion"><span class="header-section-number">14</span> User data insertion</h2>
<div class="line-block">Sometimes it is interesting to work with discrete data itself, seeking a trend, mathematical or simply visual, for the relationship between variables. Here is an example of an interactive graph with user data.</div>
<p><br>
</p>
<p><a href="Eq/jsp_dados.html" target="_blank"><img src="Eq/dados.png" class="img-fluid"></a></p>
<section id="suggestion-13" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-13">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try changing the data entered, overlaying the graph or not;</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Try changing the data representation <span class="cf">in</span> <span class="st">"mode"</span> and <span class="st">"type"</span> to points, lines, points<span class="sc">+</span>lines, bars.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
</section>
<section id="uploading-files-for-analysis" class="level2" data-number="15">
<h2 data-number="15" class="anchored" data-anchor-id="uploading-files-for-analysis"><span class="header-section-number">15</span> Uploading files for analysis</h2>
<section id="context---csv-file" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context---csv-file">Context - CSV file</h3>
<div class="line-block">Sometimes, it is necessary to analyze results related to a set of data in a spreadsheet. For this situation, the following example illustrates the loading of two data vectors from a <em>CSV</em> (<em>comma separated value</em>) file, a format commonly used for exporting/importing in spreadsheets, and the resulting graph.</div>
<p><a href="Eq/jsp_csv.html" target="_blank"><img src="Eq/csv.png" class="img-fluid"></a></p>
<p><strong>Instructions:</strong></p>
<ul>
<li>1 Click on <em>add plot</em> and select a CSV file using the <em>browse</em> button at the top. Note: variable <em>X</em> in the first column of the file, and variable <em>Y</em> in the second column;</li>
<li><ol start="2" type="1">
<li>Click on <em>add plot</em> again to view the resulting graph.</li>
</ol></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try other CSV files;</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Vary the aspects of the graph, such as type, color, marker size, etc.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="linear-data-adjustment" class="level2" data-number="16">
<h2 data-number="16" class="anchored" data-anchor-id="linear-data-adjustment"><span class="header-section-number">16</span> Linear data adjustment</h2>
<section id="context-13" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-13">Context:</h3>
<div class="line-block">In various experimental research situations, it is necessary to correlate the data of a dependent variable with a predictor variable using the following equation.</div>
</section>
<section id="equation-12" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-12">Equation:</h3>
<p><span class="math display">\[
y = \alpha x + \beta + \varepsilon
\]</span> <em>Where:</em></p>
<ul>
<li>y = dependent variable;</li>
<li>x = independent variable;</li>
<li><span class="math inline">\(\alpha\)</span> = slope of the adjusted line;</li>
<li><span class="math inline">\(\beta\)</span> = intercept of the adjusted line;</li>
<li><span class="math inline">\(\epsilon\)</span> = measurement error.</li>
</ul>
<div class="line-block">The parameters of <em>slope and intercept</em> can be obtained by several procedures, such as <em>least squares</em>, <em>matrix algebra</em>, or simple <em>summaries</em>. In the latter case, the calculation of parameters and error can be computed as follows:</div>
<p><span class="math display">\[
\alpha = \frac{n \sum x_i y_i - \sum x_i \sum y_i}{n \sum x_i^2 - \left( \sum x_i \right)^2}
\]</span></p>
<p><span class="math display">\[
\beta = \frac{\sum y_i - \alpha \sum x_i}{n}
\]</span></p>
<p><span class="math display">\[
\hat{y}_i = \alpha x_i + \beta
\]</span></p>
<p><span class="math display">\[
\varepsilon_i = \left| y_i - \hat{y}_i \right|
\]</span></p>
<div class="line-block">A representation for linear adjustment with arbitrary data is provided below:</div>
<p><br>
</p>
<p><a href="Eq/jsp_ajLin.html" target="_blank"><img src="Eq/ajLin.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-14" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-14">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Show the points that overlap or do not overlap the fit line. For overlap, choose <span class="st">"showPoints = true"</span> and <span class="st">"showLine = true"</span>;</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the data and perform a new fit to obtain other parameters of the line.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="polynomial-regression" class="level2" data-number="17">
<h2 data-number="17" class="anchored" data-anchor-id="polynomial-regression"><span class="header-section-number">17</span> Polynomial regression</h2>
<section id="context-14" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-14">Context:</h3>
<div class="line-block">In some situations, linear fitting does not provide an adequate trend for modeling experimental data, which leads to the use of other models, such as <em>exponential, hyperbolic, logarithmic, power, and polynomial</em>. Polynomial trends are common in natural phenomena, with mathematical treatment of data occurring by different methods, such as simple sums, least squares, or linear algebra. Here is an example of polynomial regression with adjustable degrees in the code.</div>
</section>
<section id="equation-13" class="level3" data-number="17.1">
<h3 data-number="17.1" class="anchored" data-anchor-id="equation-13"><span class="header-section-number">17.1</span> Equation</h3>
<div class="line-block">Error minimization for the polynomial curve is obtained by generalized <em>least squares regression</em> and application of the <em>Vandermonde matrix</em>, as follows.</div>
<div class="line-block">Let the vectors x=[x<span class="math inline">\(_{1}\)</span>,x<span class="math inline">\(_{2}\)</span>,…,x<span class="math inline">\(_{n}\)</span>] and y=[y<span class="math inline">\(_{1}\)</span>,y<span class="math inline">\(_{2}\)</span>,…,y<span class="math inline">\({_n}\)</span>] be the desired polynomial:</div>
<p><span class="math display">\[
y = \beta_0 + \beta_1 x + \beta_2 x^2 + \cdots + \beta_g x^g
\]</span> | Thus, the Vandermonde matrix is constructed as:</p>
<p><span class="math display">\[
X =
\begin{bmatrix}
1 &amp; x_1 &amp; x_1^2 &amp; \cdots &amp; x_1^g \\\\
1 &amp; x_2 &amp; x_2^2 &amp; \cdots &amp; x_2^g \\\\
\vdots &amp; \vdots &amp; \vdots &amp; &amp; \vdots \\\\
1 &amp; x_n &amp; x_n^2 &amp; \cdots &amp; x_n^g
\end{bmatrix}
\]</span></p>
<div class="line-block">To find the <em>coefficients <span class="math inline">\(\beta\)</span></em> that minimize the quadratic error, solve the linear system in sequence:</div>
<p><span class="math display">\[
\boldsymbol{\beta} = (X^T X)^{-1} X^T \mathbf{y}
\]</span><br>
</p>
<p><em>Where:</em></p>
<ul>
<li><em>T</em> represents the transposed matrix<br>
</li>
</ul>
<p><a href="Eq/jsp_polinomial.html" target="“_blank”"><img src="Eq/polinomial.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-15" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-15">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try degree <span class="dv">1</span> <span class="cf">for</span> the polynomial, i.e., a reduction of the treatment to linear adjustment;</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the formatting of labels, colors, sizes, etc., <span class="cf">in</span> the code;</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Overlap some adjustments, edit, and reposition the caption;</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Test the code with another data vector.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="multilinear-regression" class="level2" data-number="18">
<h2 data-number="18" class="anchored" data-anchor-id="multilinear-regression"><span class="header-section-number">18</span> Multilinear regression</h2>
<section id="context-15" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-15">Context:</h3>
<div class="line-block">Multiple linear adjustment is very useful when you want to predict behavior based on two or more predictor variables. The procedure involves a <em>design matrix</em> (or <em>projection operator</em>) similar to the <em>Vandermonde matrix</em>.</div>
</section>
<section id="equation-14" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-14">Equation:</h3>
<div class="line-block">The analytical solution for the design matrix involves solving a matrix of coefficients <span class="math inline">\(\beta\)</span> for a set of predictor values of the response vector <em>y</em>, as follows:</div>
<p><span class="math display">\[
y_i = \beta_0 + \beta_1 x_{1i} + \beta_2 x_{2i} + \cdots + \beta_p x_{pi} + \varepsilon_i
\]</span></p>
<p><span class="math display">\[
\mathbf{y} = \mathbf{X} \boldsymbol{\beta} + \boldsymbol{\varepsilon}
\]</span></p>
<p><span class="math display">\[
\hat{\boldsymbol{\beta}} = (\mathbf{X}^\top \mathbf{X})^{-1} \mathbf{X}^\top \mathbf{y}
\]</span></p>
<p><em>Where:</em> - <span class="math inline">\(\beta\)</span> = coefficient vector; - <em>y</em> = response vector; - <em>X</em> = design matrix; - <span class="math inline">\(\epsilon\)</span> = random noise</p>
<div class="line-block">Thus, predicted values and associated error are obtained by:</div>
<p><span class="math display">\[
\hat{\mathbf{y}} = \mathbf{X} \hat{\boldsymbol{\beta}}
\]</span> <span class="math display">\[
\mathbf{e} = \mathbf{y} - \hat{\mathbf{y}}
\]</span></p>
<div class="line-block">Obviously, the graphical representation for a multilinear fit is limited to two independent variables, <em>x1</em> and <em>x2</em>, in addition to the response variable (<em>3D graph</em>), although the matrix calculation allows for several predictor variables (hyperdimensional space). Tangential to the first situation, here is an example.</div>
<p><strong>Instructions</strong> - 1. Note that there is a Boolean flag (<em>showFit</em>) at the beginning of the code: <em>false</em> for data only, and <em>true</em> for the fit. - 2. You can click on <em>add plot</em> to view the data with the flag set to <em>false</em>, followed by another <em>add plot</em> with the flag set to <em>true</em>.<br>
</p>
<p><a href="Eq/jsp_multilinear.html" target="“_blank”"><img src="Eq/multilinear.png" class="img-fluid"></a><br>
</p>
<div class="line-block">Rotate the graph and observe the surface around the points. Unlike a linear fit, the points are distributed discontinuously in space.</div>
</section>
<section id="suggestion-16" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-16">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> <span class="fu">QSAR</span> (“Quantitative Structure<span class="sc">-</span>Activity Relationship”) results use multilinear adjustment analysis to identify the strength of predictor variables. Try a dataset with two <span class="fu">variables</span> (e.g., concentration, pH, compound A, B, etc.) on the internet.</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> For more predictor variables, see the code below<span class="sc">!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="context-multiple-linear-adjustment-with-3-or-more-predictor-variables" class="level3" data-number="18.1">
<h3 data-number="18.1" class="anchored" data-anchor-id="context-multiple-linear-adjustment-with-3-or-more-predictor-variables"><span class="header-section-number">18.1</span> Context: Multiple linear adjustment with 3 or more predictor variables</h3>
<div class="line-block">Although <em>JSPlotly</em> was designed for producing interactive graphs and maps using the <em>Plotly.js</em> library on top of the <em>JavaScript</em> language, it extends the application possibilities beyond graphical representations. An example of this can be found in the code below, for a <em>multilinear adjustment</em> for a situation with 4 predictor variables.</div>
<p><a href="Eq/jsp_multilin4Var.html" target="“_blank”"><img src="Eq/multilin4Var.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-17" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-17">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Try varying the number of <span class="fu">predictors</span> (xi). To do this<span class="sc">:</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>a. Without reducing, simply remove the desired <span class="fu">vector</span>(s) and correct their quantity <span class="cf">in</span> the line<span class="sc">:</span> “const X <span class="ot">=</span> <span class="fu">x1.map</span>((_, i)”;</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>b. If you want to increase, add a new vector <span class="cf">in</span> the corresponding line and update the quantity <span class="cf">in</span> the mapping of “const <span class="at">X =</span> <span class="fu">x1.map</span>((_, i)”.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="response-surface-methodology-rsm" class="level2" data-number="19">
<h2 data-number="19" class="anchored" data-anchor-id="response-surface-methodology-rsm"><span class="header-section-number">19</span> Response Surface Methodology (RSM)</h2>
<div class="line-block">A variation of multilinear adjustment involves the <em>response surface methodology</em>, commonly linear (flat) or quadratic (curvilinear), applied in modeling and optimization of experimental processes with multiple independent variables. Its objective is to fit a mathematical equation that represents the relationship between the experimental factors and the response variable, allowing behaviors to be predicted and optimal conditions to be identified.</div>
<div class="line-block">For experiments with two factors, a second-order quadratic model is commonly used, which allows surfaces with curvature to be represented. The central idea of the experiment is to select values and their codes (-1,0,1), representing, respectively, a quantity below, an average quantity, and a quantity above, for a given predictor variable.</div>
<section id="equation-15" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-15">Equation:</h3>
<div class="line-block">The general quadratic model for two factors is represented by:</div>
<p><span class="math display">\[
y = \beta_0 + \beta_1 x_1 + \beta_2 x_2 + \beta_{11} x_1^2 + \beta_{22} x_2^2 + \beta_{12} x_1 x_2 + \varepsilon
\]</span></p>
<div class="line-block">With the matrix form of the model represented by:</div>
<p><span class="math display">\[
\mathbf{y} = \mathbf{X} \boldsymbol{\beta} + \boldsymbol{\varepsilon}
\]</span></p>
<div class="line-block">And the estimation of the coefficients <span class="math inline">\(\beta\)</span> by least squares, similar to multilinear adjustment, by:</div>
<p><span class="math display">\[
\hat{\boldsymbol{\beta}} = (\mathbf{X}^\top \mathbf{X})^{-1} \mathbf{X}^\top \mathbf{y}
\]</span><br>
</p>
<p><a href="Eq/jsp_msr.html" target="“_blank”"><img src="Eq/msr.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-18" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-18">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> As suggested earlier, try using data from the literature or another source whose responses are already known, replacing the respective vectors at the beginning of the code. This allows you to compare the effectiveness of using the tool presented.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="data-smoothing---spline-and-savitzky-golay-filter" class="level2" data-number="20">
<h2 data-number="20" class="anchored" data-anchor-id="data-smoothing---spline-and-savitzky-golay-filter"><span class="header-section-number">20</span> Data smoothing - Spline and Savitzky-Golay filter</h2>
<div class="line-block">In experimental research, signal detection occasionally occurs in the presence of significant levels of noise, as occurs in continuous spectral scanning under wavelength variation, or in the detection of electrical signals in time. In this case, there is no analytical equation that describes the behavior of the signal, although some digital processing tools can rely on a trend curve in data processing, significantly reducing the noise level, such as by <em>spline curve</em> and <em>Savitzky-Golay filter</em>.</div>
<section id="context---cubic-spline" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context---cubic-spline">Context - Cubic spline</h3>
<div class="line-block">A very common interpolation technique involves the use of cubic splines by degree 3 polynomials interconnecting data windows. The treatment consists of applying polynomial functions to subintervals of the points, promoting smoothing between the data windows. Mathematically, the spline is applied in such a way that:</div>
<p><span class="math display">\[
S_i(x) = a_i + b_i(x - x_i) + c_i(x - x_i)^2 + d_i(x - x_i)^3
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>S<span class="math inline">\(_{i}\)</span>(x) = cubic spline function in the interval <em>[xi,xi+1]</em>;</li>
<li>a<span class="math inline">\(_{i, b\)</span><em>{i}<span class="math inline">\(, c\)</span></em>{i}<span class="math inline">\(, d\)</span>_{i}$ = specific coefficients for segment <em>i</em>;</li>
<li>x<span class="math inline">\(_{i}\)</span> = starting point of the interval;</li>
<li><em>x</em> = independent variable.</li>
</ul>
</section>
<section id="context---savitzky-golay-filter" class="level3" data-number="20.1">
<h3 data-number="20.1" class="anchored" data-anchor-id="context---savitzky-golay-filter"><span class="header-section-number">20.1</span> Context - Savitzky-Golay filter</h3>
<div class="line-block">The Savitzky-Golay filter smoothing technique applies a local polynomial regression to a moving window, using least squares and a predetermined degree, preserving peak, valley, and band signals. Unlike cubic splines, the filter produces a continuous global curve, rather than a set of interconnected curves. Mathematically:</div>
<p><span class="math display">\[
\tilde{y}_i = \sum_{j=-m}^{m} c_j \cdot y_{i+j}
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>y<span class="math inline">\(_{i}\)</span> = smoothed value at position <em>i</em>;</li>
<li>y<span class="math inline">\(_{i+j}\)</span> = actual values of the series within the window;</li>
<li>c<span class="math inline">\(_{i}\)</span> = filter coefficients derived from a polynomial regression;</li>
<li><em>m</em> = number of points on each side of the central window.</li>
</ul>
<div class="line-block">Here is an example of code for <em>JSplotly</em> that illustrates the use of the two interpolation methods.</div>
<p><a href="Eq/jsp_splineGolay.html" target="“_blank”"><img src="Eq/splineGolay.png" class="img-fluid"></a><br>
</p>
<p><strong>Instructions:</strong></p>
<ul>
<li><ol type="1">
<li>The code has two <em>true/false flags</em>, one for ‘useGolay’ and another for “useSpline”;</li>
</ol></li>
<li><ol start="2" type="1">
<li>View the raw data by setting both flags to <em>false</em>;</li>
</ol></li>
<li><ol start="3" type="1">
<li>To overlay a cubic spline, change its constant to <em>true</em>;</li>
</ol></li>
<li><ol start="4" type="1">
<li>To overlay the Savitzky-Golay filter, change its constant to <em>true</em>;</li>
</ol></li>
<li><ol start="5" type="1">
<li>To adjust the filter, modify its parameters in the constants <em>window</em> (moving window) and/or <em>degree</em> (degree of the polynomial).</li>
</ol></li>
</ul>
</section>
<section id="suggestion-19" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-19">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> Compare the smoothing effects of the two polynomial interpolation treatments;</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Apply only the filter, adjusting the moving window and polynomial degree parameters;</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Try interpolation with other raw data <span class="cf">in</span> the code, replacing the constants “x_values” and “y_raw” with numeric vectors;</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="anova-and-tukey-hsd-test" class="level2" data-number="21">
<h2 data-number="21" class="anchored" data-anchor-id="anova-and-tukey-hsd-test"><span class="header-section-number">21</span> ANOVA and Tukey-HSD test</h2>
<section id="context-16" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-16">Context:</h3>
<div class="line-block">In research, it is common to analyze a set of data by comparing the variance between groups with that obtained within each group, or <em>Analysis of Variance (ANOVA)</em>. Once statistical information indicating that there is a difference between groups greater than the internal difference has been obtained, a <em>post hoc</em> test is performed to compare means, such as the <em>Tukey-HSD test</em> (<em>Honest Significant Difference</em>).</div>
</section>
<section id="equation-16" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-16">Equation:</h3>
<div class="line-block">The equation that characterizes a Tukey-HSD test is provided below (correction for unbalanced groups):</div>
<p><span class="math display">\[
\text{HSD}_{ij} = q_{\alpha} \cdot \sqrt{\frac{\text{MS}_{\text{within}}}{2} \left( \frac{1}{n_i} + \frac{1}{n_j} \right)}
\]</span></p>
<p><em>Where:</em></p>
<ul>
<li>HSD = minimum value of the difference between two means for it to be considered significant;</li>
<li>q<span class="math inline">\(_{\alpha}\)</span> = critical value of the studentized distribution <em>q</em> (Tukey distribution), depending on the number of groups and the significance level <span class="math inline">\(\alpha\)</span>;</li>
<li>MS<span class="math inline">\(_{within}\)</span> = mean square within groups (error) obtained from ANOVA;</li>
<li>n: number of observations per group (<em>i</em> and <em>j</em> if there are groups of different sizes).<br>
</li>
</ul>
<p><a href="Eq/jsp_tukey.html" target="_blank"><img src="Eq/tukey.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-20" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-20">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> You can click and drag the minimum difference <span class="fu">markers</span> (a, b, etc.) above the graph <span class="cf">for</span> better visual adjustment, as well as the ANOVA table and the statistical test result;</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the values of the vectors, redo the analysis and the graph simultaneously, and notice the difference <span class="cf">in</span> the values obtained. Note that, as stated at the bottom, you need to refresh the website page to clear previous data retained <span class="cf">in</span> the cache;</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fl">3.</span> Enter other data <span class="cf">in</span> the respective vectors, either your own or from another source;</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="fl">4.</span> Enter new data vectors, or delete some, <span class="cf">for</span> a customized calculation and graph.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="nonlinear-data-adjustment" class="level2" data-number="22">
<h2 data-number="22" class="anchored" data-anchor-id="nonlinear-data-adjustment"><span class="header-section-number">22</span> Nonlinear data adjustment</h2>
<div class="line-block">Modeling is a common area of employment in the natural sciences, when mathematical prediction or correlation of experimental data is desired. Models can be linear or nonlinear. When nonlinear, they can be applied empirically to the data (e.g., quadratic, logarithmic, exponential, or power functions), or from an explanatory mathematical function of the model.</div>
<div class="line-block">In this case, techniques for nonlinear adjustment of the data to the model are used, such as <em>Simplex</em>, <em>Gauss</em>, <em>Newton-Raphson</em>, <em>Levenberg-Marquardt</em>, or <em>least squares</em>, for example. Regardless of the method, nonlinear adjustment differs from linear adjustment in its assumptions and treatment of data (see a quick explanation <a href="https://bioquanti.netlify.app/pt/nivel/superior/r/enzimas#ajuste-n%C3%A3o-linear">here</a>.</div>
<section id="context-ligand-protein-interaction-grid-search-algorithm" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context-ligand-protein-interaction-grid-search-algorithm">Context: Ligand-protein interaction (<em>grid search</em> algorithm)</h3>
<div class="line-block">The following example illustrates the application of a nonlinear adjustment of molecular interaction data using the Langmuir model and treatment by <em>manual grid search</em>. In this treatment, the <em>sum of the quadratic errors (RMSE)</em> is calculated for combinations of <em>a</em> and <em>b</em>, selecting the pair with the lowest calculated error. In this case, for simplicity, although <em>JSPlotly</em> works with more specific libraries, the nonlinear regression is performed by least squares, but using <em>grid search</em> rather than classic iterative methods. Thus, a set of values is tested for the pair (a, b) in a previously defined grid (combination table), allowing the identification of the parameters that best describe the experimental data according to the minimum error criterion.</div>
<p><br>
</p>
</section>
<section id="equation-17" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="equation-17">Equation:</h3>
<div class="line-block">The Langmuir equation is represented below:</div>
<p><span class="math display">\[
Y = \frac{aX}{K_d + X}
\]</span> <em>Where:</em></p>
<ul>
<li>Y = fraction of occupied sites;</li>
<li>X = concentration of free ligand;</li>
<li><em>a</em> = saturation value (sites of the same affinity completely occupied);</li>
<li><em>Kd</em> = dissociation equilibrium constant of the complex.</li>
</ul>
<div class="line-block">In this case, the nonlinear least squares function is:</div>
<p><span class="math display">\[
SSR(a, K_d) = \sum_{i=1}^n \left( y_i - \frac{a x_i}{K_d + x_i} \right)^2
\]</span></p>
<p><em>Where:</em></p>
<p><em>SSR</em> = sum of the squares of the residuals (inversely proportional to the quality of the fit)<br>
</p>
<p><a href="Eq/jsp_ajNonLin.html" target="“_blank”"><img src="Eq/ajNonLin.png" class="img-fluid"></a></p>
</section>
<section id="suggestion-21" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="suggestion-21">Suggestion:</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fl">1.</span> As <span class="cf">in</span> the previous example, alternately display the points and the adjustment curve using the “Boolean” <span class="fu">operation</span> (“<span class="at">showPoints =</span> true<span class="sc">/</span>false”; “<span class="at">showCurve =</span> true<span class="sc">/</span>false”);</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fl">2.</span> Change the data and perform a new fit to obtain other parameters of the <span class="fu">equation</span> (Kd, SSR, R², adjusted R²). Note<span class="sc">:</span> R² adj corresponds to the value of R² corrected <span class="cf">for</span> the number of parameters <span class="cf">in</span> the <span class="fu">model</span> (<span class="cf">in</span> this case, “<span class="dv">2</span>”), directly affecting the “degrees of freedom” <span class="cf">for</span> the fit.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><br>
</p>
</section>
<section id="context---competitive-inhibition-in-enzyme-kinetics-gauss-newton-algorithm" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="context---competitive-inhibition-in-enzyme-kinetics-gauss-newton-algorithm">Context - Competitive inhibition in enzyme kinetics (Gauss-Newton algorithm)</h3>
<div class="line-block">The <em>Gauss-Newton</em> method iteratively fits nonlinear models to experimental data sets. It is especially efficient when the model is <em>quasi-linear</em> with respect to the parameters and the residuals are moderately small. The goal is to find the parameters <span class="math inline">\(\theta\)</span> that minimize the sum of the squares of the residuals between the observed data <em>yi</em> and the values predicted by the model f(xi,<span class="math inline">\(\theta\)</span>. Unlike the full Newton-Raphson method, Gauss-Newton avoids the explicit calculation of the Hessian by using an approximation based on the Jacobian of the residuals.</div>
</section>
</section>
<section id="equation-18" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="equation-18">Equation:</h2>
<div class="line-block">This approximation results in a linear system that can be solved at each iteration to update the model parameters. The central formula of the method consists of multiplying the transposed Jacobian <em>JT</em> by the Jacobian <em>J</em> itself, forming a symmetric matrix that approximates the Hessian. The residual vector <em>r</em> is then used to determine the update direction <span class="math inline">\(\Delta\theta\)</span>, with the following equation:</div>
<p><span class="math display">\[
\theta_{k+1} = \theta_k + \left(J^T J\right)^{-1} J^T \cdot r
\]</span></p>
<div class="line-block">To illustrate the potential of the application in iterative nonlinear fitting, here is an example for enzyme kinetics using a <em>competitive inhibition model</em>.</div>
<div class="line-block">The code allows you to adjust the <em>seeds</em>, <em>tolerance</em>, and number of <em>iterations</em>, and provides the graph (points/fitted line, manually), and statistical parameters of the regression.</div>
<p><br>
</p>
<p><strong>Instructions</strong></p>
<ul>
<li><ol type="1">
<li>Note that the code has a flag for <em>true/false</em> in the <em>ajuste</em> constant. I left it at <em>false</em> for data representation only (vectors <em>x</em> and <em>y</em>, or <em>S</em> and <em>v</em>);</li>
</ol></li>
<li><ol start="2" type="1">
<li>In the <em>Plot</em> area of the code, set the <em>trace</em> constant so that it also includes the data vectors;</li>
</ol></li>
<li><ol start="3" type="1">
<li>Click <em>add plot</em> to view the experimental points;</li>
</ol></li>
<li><ol start="4" type="1">
<li>To superimpose the fitted curve, update the constants <em>ajuste</em> and <em>trace</em> (vectors <em>S_fit</em> and <em>v_fit</em>), and choose <em>lines</em> in <em>mode</em>.</li>
</ol></li>
</ul>
<p><a href="Eq/jsp_ajNonLinGauss.html" target="“_blank”"><img src="Eq/ajNonLinGauss.png" class="img-fluid"></a></p>


</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../../../nivel/superior/jsplotly/gsplotly_en.html" class="pagination-link" aria-label="GSPlotly - AI code generator for JSPlotly">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">GSPlotly - AI code generator for JSPlotly</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../../nivel/basico/jsplotly/jsplotly_bas_en.html" class="pagination-link" aria-label="Simulations at School">
        <span class="nav-page-text">Simulations at School</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>